\documentclass[10pt,a4paper]{article}
\usepackage[utf8]{inputenc}

\pdfoutput=1
 
%~ \usepackage{cite}
%~ \makeatletter
%~ \renewcommand{\@biblabel}[1]{\quad#1.}
%~ \makeatother


\usepackage{nameref}
\usepackage[colorlinks=true,citecolor={Gray}]{hyperref}
%~ \usepackage[right]{lineno}

\usepackage{url}

\usepackage[top=0.85in,left=2.75in,footskip=0.75in,marginparwidth=2in]{geometry}

% headrule, footrule and page numbers
\usepackage{lastpage,fancyhdr,graphicx}
\usepackage{epstopdf}
\pagestyle{myheadings}
\pagestyle{fancy}
\fancyhf{}
\rfoot{\thepage/\pageref{LastPage}}
\renewcommand{\footrule}{\hrule height 2pt \vspace{2mm}}
\fancyheadoffset[L]{2.25in}
\fancyfootoffset[L]{2.25in}

\raggedright
\setlength{\parindent}{0.5cm}
\textwidth 5.0in 
\textheight 8.75in

\usepackage{changepage}

\usepackage{microtype}
\DisableLigatures[f]{encoding = *, family = * }

\usepackage{color}
\definecolor{Gray}{gray}{.25}


\usepackage{graphicx}
\usepackage{sidecap}
\usepackage{wrapfig}
\usepackage[pscoord]{eso-pic}
\usepackage[fulladjust]{marginnote}
\reversemarginpar

\usepackage[natbib=true, style=authoryear, uniquename=false, maxcitenames=2, maxbibnames=99, firstinits=true, doi=false, url=false, backend=bibtex]{biblatex}
\bibliography{robustness.bib}
\renewbibmacro{in:}{}                                  % remove In:
\DeclareFieldFormat[article, inbook]{title}{#1}        % remove quotes around titles
\AtEveryBibitem{%                                      % remove language indications
  \clearlist{language}%
}

\usepackage{bm}
\usepackage{amsmath}

\usepackage[english]{babel}

\usepackage[page,title]{appendix}
\addto\captionsenglish{%
	\renewcommand{\appendixname}{Appendix}
	\renewcommand{\appendixpagename}{}%Supplementary material}
}

\widowpenalty10000
\clubpenalty10000

% Replacements in the text

\newcommand{\stability}{{\rho_S}}
\newcommand{\earlyenv}{{\rho_E}}
\newcommand{\lateenv}{{\rho_e}}
\newcommand{\earlymut}{{\rho_M}}
\newcommand{\latemut}{{\rho_m}}

\newcommand{\W}{\bm{\mathrm W}}
\newcommand{\M}{\bm{\mathrm M}}
\newcommand{\Pp}{\bm{\mathrm P}}

\newcommand{\SupMat}{Appendix~}

\title{}
\date{}

\begin{document}
\vspace*{0.35in}

% title goes here:
\begin{flushleft}
{\Large
\textbf\newline{Gene network robustness as a multivariate character}
}
\newline
% authors go here:
\\
Arnaud Le~Rouzic\textsuperscript{1,*}
\\
\bigskip
\bf{1} Laboratoire Évolution: Génomes, Comportement, Écologie; Université Paris-Saclay, CNRS, IRD.
\\
\bigskip
* arnaud.le-rouzic@egce.cnrs-gif.fr

\end{flushleft}

\section*{Abstract}

Robustness to genetic or environmental disturbances is often considered as a key property of living systems. Yet, in spite of being discussed since the 1950s, how robustness emerges from the complexity of genetic architectures and how it evolves still remains unclear. In particular, whether or not robustness is independent to various sources of perturbations conditions the range of adaptive scenarios that can be considered. For instance, selection for robustness to heritable mutations is likely to be modest and indirect, and its evolution might result from indirect selection on a pleiotropically-related character (e.g., homeostasis). Here, I propose to treat various robustness measurements as quantitative characters, and study theoretically, by individual-based simulations, their propensity to evolve independently. Based on a simple evolutionary model of a gene regulatory network, I showed that different ways to measure the robustness of gene expression to genetic or non-genetic disturbances were substantially correlated. Yet, robustness was mutationally variable in several dimensions, and robustness components could evolve differentially under direct selection pressure. Therefore, the fact that the sensitivity of gene expression to mutations and environmental factors rely on the same gene networks does not preclude distinct evolutionary histories of robustness components.

%~ \linenumbers

\section{Introduction}

Robustness is the capacity of living organisms to buffer internal or environmental disturbances. Robustness encompasses, for instance, the ability to maintain physiological equilibria (homeostasis), to ensure developmental stability, or to repair and mitigate DNA damage in both soma and germline. Although robustness is virtually intermingled with the definition of life itself, its underlying mechanisms and its evolutionary origins remain far from being clearly understood \citep{Ste02,MS09,Wag13,HGK+19}. 

Robustness evolves as a consequence of non-linearities in the developmental or physiological mechanisms, i.e.\ changes in the magnitude of the effect of some genetic or environmental factor on the phenotype of interest \citep{Nij02}. The study of the evolutionary processes leading to robustness roots into the conceptual and empirical work by C.H.\ Waddington and the concept of canalization \citep{Wad42,Sch49,Wad59,Loi19}. Canalization is a property of complex developmental systems that buffers environmental and genetic variation, and maintains actively the organism in an optimal developmental path. Although the scope and the definition of canalization varies substantially among authors, canalization is generally expected to evolve as an adaptation to "canalizing" selection for an optimal phenotype \citep{EM98,DD01,Fla05,Kli19}. However, formal population genetic models have questioned the unicity of the canalization process. In particular, robustness to environmental factors appears more likely to evolve as an adaptation than robustness to genetic (mutational) disturbances, on which selection seems to be rather weak and indirect even in optimal theoretical conditions \citep{WBB97, HHW03,LAH13}. 

In this context, the evolution of robustness as a general property of organisms heavily depends on the genetic and physiological integration of the different robustness dimensions \citep{Far15}. If the robustness to environmental factors and to genetic mutations share the same physiological bases, the adaptive evolution of environmental canalization can generate a correlated response of genetic canalization; this hypotheses has been refered to as "congruent evolution" \citep{dHW+03}. In contrast, if genetic and environmental robustness had independent biological bases, they would be featured by independent evolutionary mechanisms, and possibly independent evolutionary histories. 

Although this issue would benefit from a better theoretical framework, modelling the evolution of robustness is not straightforward. The simplest approach relies on focusing on modifiers, i.e. genes that can influence the robustness of the organism without affecting the phenotype. However, in the case of genetic robustness, modifier-based models either rely on tricky rescaling or cannot dissociate the phenotype and the robustness to the phenotype \citep{WBB97, Kaw00, RM13}. In addition, in models where the genotype-phenotype association is arbitrary, any correlation between environmental and genetic robustness is a modelling choice, and not an output of the model. More promising to address the congruent evolution issue are models in which the phenotype is a result of an integrated process mimicking some developmental or physiological mechanism (refered to as \emph{causally cohesive
genotype phenotype} models in \cite{RGV08}). In such dynamic models, robustness to various disturbances appear as an emergent property of the model complexity, caused by regulatory feedbacks, that cannot be easily deduced from the model parameters. Although the potential palette of relevant dynamic models is large and could include morphological development models \citep{MS20} or metabolic models \citep{NBR19}, evolutionary biologists have often considered gene regulatory network models as a good compromise between complexity and numerical tractability for studying the evolution of canalization and robustness \citep{Kau69,Wag94,SBB00,LP12}.

Such theoretical gene networks have been shown to display enough non-linearity, leading to epistasis and pleiotropy, to evolve enhanced or reduced sensitivity to environmental \citep{Mas04,EMW11,EMW11b} and genetic \citep{Wag96,BS03,DW08,ALS+06,RL16} perturbations. Interesting observations suggest that environmental or genetic canalization could be correlated to other robustness properties in such models. For instance, \citet{CMW07} noticed that robustness to mutations and robustness to noise was correlated. Furthermore, it has been shown that network stability, the propensity of the network to maintain stable (non-cyclic) gene expressions, was correlated to robustness, as selection on stability alone could drive an indirect response of genetic \citep{SB02} and environmental \citep{Mas04} canalization. In contrast, \citet{ORL18} showed that networks selected to maintain (but not converge to) an equilibrium became both environmentally sensitive and genetically canalized, suggesting that environmental and genetic robustness could be theoretically decoupled. However, no systematic quantitative description of the pleiotropic pattern underlying different robustness components has ever been attempted. 

Here, I aim at extending the study of canalization in theoretical gene networks to address the multidimensional nature of robustness, by estimating the evolutionary independence of various robustness components. Four robustness-related measurements were considered, two of them corresponding to environmental robustness (early vs.\ late disturbances), two corresponding to genetic robustness (early --- inherited --- or late --- acquired --- mutations). Gene expression instability was also included in the set of robustness-related traits, as it is related to the intrinsic stability of the expression phenotype. The first part of this study focuses on the multidimensional patterns of robustness in small and random networks, and the second part on the evolutionary consequences of the pleiotropic nature of robustness, based on individual-based simulations. 

\section{Model and Methods}

\subsection{Gene regulatory network}

The network model belongs to the family of gene regulatory network models sometimes refered to as "Wagner model" (after \citealp{Wag94,Wag96}; see \citealp{FP15} for a historical record), with two main differences: (i) the network output (gene expressions) is quantitative and not qualitative, in the same way as in \citet{SB02}, and (ii) the expression of unregulated genes (constitutive expression) was lower than half the maximum expression. 

More specifically, the structure of a $n$-gene network is encoded as a $n\times n$ matrix $\W$, while the state of the network is stored into a vector of size $n$, $\Pp$. In this setting, $W_{ij}$ encodes the influence of gene $j$ on the expression of gene $i$, $W_{ij} < 0$ represents a negative interaction (inhibition), $W_{ij} > 0$ a positive interaction (activation), and $W_{ij} = 0$ denotes the absence of regulatory interaction. $P_i$ is the expression of gene $i$, ranging between 0 (no expression) and 1 (maximum expression). 

The properties of these gene networks are explored in a discrete dynamic system:
\begin{equation}
 \Pp_{t+1} = F(\W \Pp_t),
\end{equation}
\noindent where the function $F$ is a vectorized version of a sigmoid scaling function: $F(x_1, x_2, \dots, x_n) = [f(x_1), f(x_2), \dots, f(x_n)]$;
\begin{equation} \label{eq:fx}
f(x) = \frac{1}{1+ \lambda_a e ^{- \mu_a x}}, 
\end{equation}
\noindent with $\lambda_a = (1-a)/a$ and $\mu_a = 1/a(1-a)$ \citep{GCL+18}. The function $f$ is scaled such that $f(0) = a$ and $df/dx|_{x=0}=1$; the parameter $a$ thus stands for the constitutive gene expression (the expression of a gene in absence of regulators), and this function defines the scale of the matrix $\W$: $W_{ij} = \delta$ ($\delta \ll 1$) means that the expression of gene $i$ at the next time step will tend to $P_{i,t+1} = a + \delta$ if $i$ is regulated by a single, fully expressed transcription factor $j$ ($P_{j,t} = 1$).

Gene networks dynamics start from an initial expression $\Pp_0$, and gene expression was updated for $T$ timesteps. By default, $\Pp_0 = (a, a, ..., a)$, since this step immediately follows a virtual initial state with no expression. The expression phenotype corresponding to a gene network was determined by averaging gene expressions during the last $\tau$ timesteps for each gene $i$: $P_i^* = (1/\tau)\sum_{t=T-\tau}^T P_{it}$. 

\subsection{Robustness indicators}

Five robustness indicators were calculated, corresponding to five different aspects of genetic or environmental robustness in a gene network: robustess to early ($\earlyenv$) and late ($\lateenv$) environmental disturbance, and robustness to early ($\earlymut$) and late ($\latemut$) genetic disturbance, and network stability $\stability$. All indicators were expressed on a scale homogeneous to log variances in gene expressions; the mode of calculation is summarized in Table~\ref{tab:indicators}, robustness is maximal when the indicator $\rho$ is small. 

The robustness to early environmental disturbance $\earlyenv$ measures the capacity of a network to reach a consistent final state starting from different initial gene expressions. In practice, $R$ replicates of the network dynamics were run, in which the initial gene expressions ($\Pp_0$) were drawn into Gaussian ($\mu = a, \sigma = \sigma_E$) distributions (expression values $<0$ and $>1$ were set to $0$ and $1$, respectively). The environmental robustness $\earlyenv_i$ for each gene $i$ was measured as the log variance in the final gene expression across these replicates. 

The robustness to late environmental disturbance $\lateenv$ measures the capacity of a network to recover its equilibrium state after having being disturbed. Gene expressions after $T$ timesteps were disturbed by adding a random Gaussian noise of standard deviation $\sigma_e$ to each gene of the network, and $\lateenv_i$ was computed for each gene $i$ as the log variance in gene expression at time step $T+1$ over $R$ replicates. 

The robustness to early mutations $\earlymut$ measures the system robustness to inherited genetic mutations (modifications of the $\W$ matrix). A random non-zero element of the $\W$ matrix was shifted by a random Gaussian number of standard deviation $\sigma_M$, and its consequences on the mean expression of all network genes was recorded. The procedure was replicated $R$ times, and the robustness score $\earlymut_i$ for each gene $i$ was calculated as the log variance of gene expression across $R$ replicates. 

The robustess to late mutations $\latemut$ measured the effect of mutations in the gene network $\W$ after having reached the final state. In practice, the $\W$ matrix was mutated in the same way as for $\earlymut$ with a standard deviation $\sigma_m$, but its consequences on gene expression were calculated for only one timestep, starting from the last state of the network. The robustness score was calculated as for other indicators (log variance over $R$ replicates).

Finally, dynamic systems based on the Wagner model often tend to generate limit cycles and never converge to a stable equilibrium. Network stability $\stability$ quantifies the capacity for a specific network to lead to stable gene expressions.  For consistency with other indicators, this instability was measured as the log squared difference between the average expression during the last $\tau$ timesteps, and an extra timestep. 

\begin{table}
\begin{adjustwidth}{-1in}{0in}
\begin{flushright}
\begin{tabular}{lp{3.5cm}ll}
Indicator & Robustness component & Computation & Disturbance std.\ dev. \\ \hline
$\earlyenv$ & \raggedright  Early noise in gene \mbox{expression} & $\earlyenv_i = \log [ \frac{1}{R-1} \sum_{r=1}^R (P_{i,r}^* - \overline{P_{i}^*})^2 ]$ & $\sigma_E=0.1$\\
$\lateenv$  & \raggedright  Late noise in gene \mbox{expression} & $\lateenv_i = \log [ \frac{1}{R-1} \sum_{r=1}^R (P_{i,T+1,r} - \overline{P_{i,T+1}})^2]$ & $\sigma_e=0.1$\\
$\earlymut$ & \raggedright  Early (inherited) \mbox{mutations} & $\earlymut_i = \log[ \frac{1}{R-1} \sum_{r=1}^R (P_{i,r}^* - \overline{P_{i}^*})^2]$ & $\sigma_M = 0.1$\\
$\latemut$ & \raggedright  Late (aquired) \mbox{mutations} & $\latemut_i = \log [ \frac{1}{R-1} \sum_{r=1}^R (P_{i,T+1,r} - \overline{P_{i,T+1}})^2]$ & $\sigma_m = 0.1$ \\
$\stability$ & \raggedright Expression stability & $\stability_i = \log[( P_i^* - P_{T+1})^2]$ & \\
\end{tabular}
\caption{\color{Gray}  \label{tab:indicators} Summarized calculation of all five robustness indicators. Index $i$ stands for the gene ($1 \leq i \leq n$), and $r$ for the replicate ($1 \leq r \leq R$), since all indicators except $\stability$ were estimated by a resampling procedure. $P_i^*$ stands for the equilibrium gene expression of gene $i$ (mean expression from the last $\tau$ timesteps), and $\overline{P_{i}^*} = (1/R)\sum_{r=1}^R P_{i,r}^*$ represents the mean over replicates. Noise in gene expression was simulated by adding a random Gaussian deviation to the initial state $\Pp_0$ of the network (for $\earlyenv$) or to the last state $\Pp_T$ of the network (for $\lateenv$). Mutations were simulated by adding a random deviation to a random interaction in the network $\bm{\mathrm W}$, either before starting the network dynamics ($\earlymut$) or after the last time step ($\latemut$). All robustness indicators are homogeneous to a log variance in gene expression; robustness increases when the indicator gets smaller, and sensitivity increases when the indicator increases. The last column indicates the standard deviation of the corresponding Gaussian disturbance. }
\end{flushright}\end{adjustwidth}
\end{table}

All these scores were calculated for every gene $i$ of a given network, and then averaged over all genes in order to get a series of summary network descriptors. The magnitude of the score itself is arbitrary, as it depends on the size of the disturbance. However, indicators happen to increase approximately linearly with the size of the disturbance (\SupMat \ref{supp:sensitmag}), the results were thus largely unaffected by a change in the variance of mutational effects and environmental noise. 

\subsection{Random networks}

Random networks were generated as $n\times n$ $\bm{\mathrm W}$ matrices filled with independent identically-distributed random numbers drawn into a Gaussian (by default: $\mu_0 = 0, \sigma_0 = 1$) distribution.  A density parameter $1/n \leq d \leq 1$ could be specified, corresponding to the frequency of non-zero slots in the $\W$ matrix. Zeros were placed randomly, with the constraint that all genes should be regulated by at least another one. 

\subsection{Exhaustive exploration of two-gene networks}

The main interest of gene-network models is the complexity and the richness of the underlying genotype-phenotype relationship. As a side effect, such models are in general difficult to handle mathematically (\cite{CTH11,LP12}). Excluding the one-gene self-regulating case (which already has non-trivial mathematical properties, \citealp{GCL+18}), the simplest network (2-by-2 matrix) has four genetic parameters, which makes the exploration of the parameter set tedious. Here, the number of dimensions was restricted by considering the set of networks that lead to a predefined arbitrary equilibrium, $\bm \Pp^{\theta}_\infty = (P^{\theta}_1, P^{\theta}_2)$. As $F(\W \Pp^{\theta}_\infty) = \Pp^{\theta}_\infty$, the $\W$ matrix can be reduced to two independent parameters, $W_{11}$ and $W_{21}$:
\begin{equation}
    \W = F \left [\begin{pmatrix} W_{11} & A \\ W_{21} & B \end{pmatrix}  \begin{pmatrix} P^{\theta}_1 \\ P^{\theta}_2 \end{pmatrix} \right] = \begin{pmatrix}P^{\theta}_1 \\ P^{\theta}_2 \end{pmatrix},
\end{equation}
\noindent with
\begin{equation}
	\label{eq:predef}
    \begin{split}
        A = \frac{1}{P^{\theta }_2} [f^{-1}(P^{\theta }_1)-W_{11}P^{\theta }_1], \\
        B = \frac{1}{P^{\theta }_2} [f^{-1}(P^{\theta }_2) - W_{21} P^{\theta }_1],
    \end{split}
\end{equation}
\noindent $f^{-1}(y) = -\frac{1}{\mu_a} \log \left( \frac{1-y}{\lambda_a y} \right)$ being the inverse of $f(x)$ (equation~\ref{eq:fx}). This equation can be extended to any network size, provided that a single element $W_{ij}$ is unknown for each line $i$ of the matrix:

\begin{equation}\label{eq:predefN}
	W_{ij} = \frac{1}{P^{\theta}_j} [f^{-1}(P^{\theta}_i) - \sum_{j^\prime \neq j} W_{ij^\prime} P^{\theta}_{j^\prime} ].
\end{equation}

Among the $n^2$ elements of a $n$-gene network, there are thus $n(n-1)$ neutral dimensions that can be explored without modifying equilibrium gene expressions. Large gene networks are thus characterized by a proportionally larger neutral space.

The $\W$ matrix achieving the desired $\Pp^{\theta \ast}_\infty$ equilibrium from a specific pair $W_{11}, W_{21}$ always exists (and is unique), but the stability of the equilibrium is not guaranteed. Networks which final gene expression $\Pp^\ast = (P^\ast_1,P^\ast_2)$ differed substantially from the target (in practice, when $|P^\ast_1 - P^{\theta}_1| + |P^\ast_2 - P^{\theta}_2| > 0.15$) were excluded from the analysis. Such discrepancies correspond to either unstable equilibria (in which case gene expressions were driven away from the equilibrium) or extreme oscillatory behaviors (large oscillations may hit expression limits 0 or 1, which drives the average expression away from the target equilibrium). 

\subsection{Evolutionary simulations}

The evolution of gene networks under various evolutionary constraints was studied by individual-based simulations. Each individual was featured by its genotype (a $n \times n$ $\W$ matrix, by default $n=6$), its expression phenotype $\Pp^*$, and the five robustness scores $\stability$, $\earlyenv$, $\lateenv$, $\earlymut$, and $\latemut$. Individuals were haploid and reproduced clonally. Mutations consisted in adding a random Gaussian deviate of variance $\sigma_\nu^2$ to an element of the $\W$ matrix, with a rate $\nu$ per individual and per generation. Generations were non-overlapping, and population size $N$ was constant. A generation consists in sampling $N$ new individuals among the $N$ parents, with a probability propotional to the individual fitness. Fitness was computed assuming stabilizing selection around a target (optimal) expression level for $n^\prime \leq n$ genes of the network (by default $n^\prime=3$), as $w = \exp(- \sum_{i=1}^{n^\prime} s_i (P_i^* - \theta_i)^2 )$, where $s_i$ was the strength of stabilizing selection on gene $i$ ($s_i = 0$ standing for no selection), and $\theta_i$ was the optimal expression phenotype. The $\theta_i$ were drawn in a uniform (0,1) distribution at the beginning of each replicated simulation, and the initial gene network was empty ($W_{ij} = 0$) except for one random element per line, which was initialized to match the optimal expression using equation~(\ref{eq:predefN}). Simulation runs were replicated 100 times and the results were averaged out, default parameter values are provided in Table~\ref{tab:defpar}. 

\begin{table}
\begin{tabular}{lll}
Parameter & Symbol & Value \\ \hline
Population size & $N$ & $1000$ \\
Gene network size & $n$ & $6$ \\
Constitutive expression & $a$ & $0.2$ \\
Network time steps & $T$ & $16$ \\
Network measurement steps & $\tau$ & $4$ \\
Network density & $d$ & $1.0$ \\
Simulation replicates & & $100$ \\
Number of generations & $G$ & $5000$ \\
Mutation rate per individual & $\nu$ & $0.01$ \\
Size of mutational effects & $\sigma_\nu$ & $0.1$ \\
Number of selected genes & $n^\prime$ & $3$ \\
Stabilizing selection coefficient & $s$ & $10$ \\
Directional selection coefficient & $\beta$ & $0$ \\
Number of robustness tests & $R$ & $100$ \\
Size of early environmental noise & $\sigma_E$ & $0.1$ \\
Size of late environmental noise & $\sigma_e$ & $0.1$ \\
Size of early genetic mutations & $\sigma_M$ & $0.1$ \\
Size of late genetic mutations & $\sigma_m$ & $0.1$ \\
\end{tabular}

\caption{\label{tab:defpar} Default parameter values in the evolutionary simulations.}
\end{table}

Directional selection on robustness indicators was also performed in some simulations, consisting in multiplying individual fitness by $\exp ( \sum_{x \in (S, E, e, M, m)} \beta_x \rho _x )$, where $\beta_x$ was the strength of directional (positive or negative) selection on robustness index $x$ (in practice, $\beta_x= \pm 0.01$). The vector $\bm\beta$ is thus proportional to the multivariate selection gradient on robustness components. There was no correlated selection (the fitness function is the product of independent marginal functions applied on gene expressions and robustness components). 

Estimating genetic covariance matrices $\bm{\mathrm G}$ was computationally untractable in simulations (it would require a heavy resampling procedure in each individual), mutational covariances $\M$ from the average genotype in the population ($\overline{\W}$) were used instead to derive multivariate evolutionary predictions. Mutational covariance matrices $\M = \nu \bm{\mathrm C}/5$ were estimated from covariances $\bm{\mathrm C}$ in gene expressions and robustess coefficients among 100 gene networks differing from $\overline{\W}$ by 5 mutations (drawn from the same algorithm as during the simulations). In order to control for the influence of stabilizing selection on gene expression on the evolution of robustness, conditional mutational matrices (equivalent to conditional evolvabilities of $ \bm{\mathrm G}$ matrices in \cite{HH08}) were computed as $\M_{c(y|x)} =\M_y - \M_{yx} \M_x^{-1} \M_{xy}$, where $y$ indicate the $n_y$ unconstrained traits and $x$ the $n_x$ constrained traits (i.e.\ the $n^\prime = 3$ genes under stabilizing selection). $\M_{c(y|x)}$ was thus a $n_y \times n_y$ matrix measuring how the unconstrained traits can mutate while traits $x$ remain constant. Predicted mutational evolvabilities in the direction of selection $\bm \beta$ were calculated as $e_\mathrm{pred} = \bm\beta^\top \M_c \bm\beta / |\bm\beta|^2$ \citep{HH08}, and realized (observed) evolvabilities were obtained by projecting the multivariate response to selection $\bm{\mathrm R}$ on the direction of $\bm\beta$: $e_\mathrm{obs} = \bm{\mathrm R} \bm\beta / |\bm\beta|$. 

Simulations and data analysis were coded in R \citep{R20}, except for the core gene network dynamics that was coded in C++ and embedded in the R code with the Rcpp package \citep{EF11}. Scripts to reproduce simulations and figures are available as a GitHub directory (\url{https://github.com/lerouzic/robustness}). 

\section{Results}

\subsection{Random networks}

Random interaction matrices are regularly used in the literature to study the general properties of gene networks (e.g.\ \cite{CTH11,PBF12}). As such, random networks are not expected to reflect the properties of biologically-realistic genetic architectures, as biological networks are far from random. However, such an approach helps developing a general intuition about the properties of the underlying model. 

Correlations were calculated between all five robustness components over 10,000 random networks (\SupMat \ref{supp:fullcorr}). All robustness components were positively correlated, correlations ranged from about 0.62 (late genetic vs.\ early environmental) to above 0.97 (late environmental vs.\ late genetic). A Principal Component Analysis (Figure~\ref{fig:pca}A and B) confirms that robustness components were partially correlated. The first PC (82\% of the total variance) corresponds to the general robustness of the network, and involves all robustness indexes. The remaining variance is explained by orthogonal vectors separating all other robustness components. At least 4 out of 5 PCs, explaining 10\% to 2\% of the total variance, did not vanish when increasing the sample size (\SupMat \ref{supp:PC}). The part of the variance in robustness explained by the first PC is robust to the way random networks were generated, as it remains around 80\%  when the mean and the variance in the regulation strenghts, the network density, and the network size vary (Figure~\ref{fig:pca}C, D, E, and F). 

\begin{figure}[t]
\begin{adjustwidth}{-1.5in}{0in}
\includegraphics{results/fig1}
\caption{\color{Gray} \label{fig:pca} A: Summary of the principal component analysis on the five robustness indicators over 10,000 random 6-gene networks ($\mu_0=0, \sigma_0=1$), indicating the position of the five robustness components on all five (normalized) Principal Components (PC); $\stability$: Stability, $\earlyenv$: Early environmental, $\lateenv$: Late environmental, $\earlymut$: Early genetic, $\latemut$: Late genetic. B: relative contribution of the five PCs to the total variance. C: Influence of the average regulation strenght ($\mu_0$) on the \% of the total variance explained by the first PC (negative values feature inhibitory networks, positive values activating networks). D: Influence of the standard deviation of the regulation strength ($\sigma_0$). E: Influence of the network density. F: Influence of the network size. }
\end{adjustwidth}
\end{figure}

\subsection{Two-gene networks}

In the following, I considered an arbitrary case of a two-gene network which genes are expressed to $\Pp_\infty$ = (0.3, 0.6). Equivalent results could be achieved with a different, arbitrary target. Figure~\ref{fig:imgpanels} illustrates how the robustness components varied in this constrained 2-gene network model (red stands for maximum robustness, i.e.\ minimum scores for $\stability$, $\earlyenv$, $\lateenv$, $\earlymut$, and $\latemut$). All the networks considered here converge to the same gene expression, and can thus be considered as phenotypically equivalent ; the colored space in Figure~\ref{fig:imgpanels} thus represents a connected neutral network in which populations can evolve, and thus change the topology and the robustness of the gene network, while keeping the expression phenotype constant. In the white regions, the equilibrium was not achieved in numerical simulations for at least three different reasons (\SupMat \ref{supp:whyitfails}): (i) fluctuations around the equilibrium were large enough to hit the edges of the (0,1) interval, shifting the mean expression; (ii) the expression dynamics was slow and the network was unable to get close to the equilibrium after 16 time steps; (iii) the equilibrium was not reachable from the default starting point.

\begin{figure}[t]
\begin{adjustwidth}{-1.5in}{0in}
\begin{flushright}
\includegraphics{results/fig2}
\caption{\color{Gray} \label{fig:imgpanels} Robustness indicators ($\earlyenv$, $\lateenv$, $\earlymut$, $\latemut$, and $\stability$) estimated for an exhaustive continuum of two-gene networks with an arbitrary expression equilibrium at $\Pp_\infty = (0.3, 0.6)$. Although two-gene networks have four independent genetic parameters, only two were represented here, the two others being computed to ensure the desired equilibrium. Red stands for the maximum robustness (lowest robustness scores); yellow for minimum robustness (highest scores). For readability, color scales are different across panels. Letters A to E stand for five example networks illustrated in \SupMat \ref{supp:simpanels}.}
\end{flushright}\end{adjustwidth}
\end{figure}

The different robustness components were correlated, but did not overlap perfecty. In order to assess the variation of the robustness properties, five networks of contrasted robustness, labeled from A to E, were tracked more specifically (Figure~\ref{fig:imgpanels}; the corresponding $\W$ matrices are provided in \SupMat \ref{supp:W}). \SupMat \ref{supp:simpanels} illustrates the effect of various sources of disturbance on each network dynamics. The network denoted as~B was robust to most sources of disturbance, while network~E was sensitive to all components except stability. Network~C was unstable, but remained relatively buffered. Networks~A and~D illustrate intermediate loss-of-robustness behaviors, through different mechanisms (instability for network~D, and weak buffering for network~A). 

This 2-gene network analysis thus confirms the results obtained for large random networks: robustness components are only partially correlated. Robustness is not a feature of large and intricate genetic architectures, as it is already present (and multidimensional) in the simplest gene networks. 

\subsection{Evolution and evolvability of robustness}

The evolution of robustness was studied by individual-based simulations, in which all individuals were characterized by their genotype (a 6-gene network) and a set of phenotypes (gene expressions and network robustness). Gene expressions for 3 out of 6 genes were under stabilizing selection. In addition to stabilizing selection on gene expression (forcing the network to maintain a functional role), robustness indicators were directly selected towards more or less sensitivity. Stabilizing selection on gene expression is expected to generate a slight selection pressure on the robustness, but this effect was apparent only for larger or more frequent mutations (\SupMat~\ref{supp:explo}). 

Direct selection on all robustess components lead to a response, showing that robustness is evolvable (diagonal panels in Figure~\ref{fig:evol}). Yet, the evolutionary potential differed substantially among robustness indicators, as indicated by the differences in the Y-scales. Robustness indicators being all homogeneous to a sum of squared difference in gene expression (i.e., the variance in gene expression induced by various disturbances), they could be compared directly. The most evolvable robustness components were early environmental disturbances ($\earlyenv$) and stability ($\stability$), which can differ by up to 25 log units (11 orders of magnitude) after 10,000 generations of bidirectional selection. In contrast, robustness to late environmental noise $\lateenv$ and genetic changes ($\earlymut$ and $\latemut$) only differed by 3 to 4 log units (i.e.\ a factor 10 to 100). For these three robustness components ($\lateenv$, $\earlymut$, and $\latemut$), the response was clearly asymmetric (the response towards more robustness was slower). Although the average response supports a clear evolutionary trend, response to selection was variable across simulation replicates, as distributions of up and down responses generally overlap. The selection response was still ongoing after 10,000 generations. 

Selection on robustness components also lead to an indirect response of all other components, which confirms a general genetic correlation. The magnitude of the correlated response (from 10\% to 100\% of the direct response) depended on the correlation across robustness components. Simulations were run to test the long-term effect of synergistic and antagonistic selection on all pairs of robustness indicators (Figure~\ref{fig:evolvability}), and selection responses after 1,000 generations were compared to the mutational evolvabilities computed at the beginning of the simulations. There was a convincing proportional relationship between predicted and observed evolvabilities on all directions of selection. Selection response was fast in directions that were mutationally evolvable, and slow in directions that were not evolvable. Yet, in spite of the variation of evolvability across directions in the multivariate robustness space, evolution was always possible, even if reduced proportionally to the mutational variance, confirming the absence of absolute constraints. 

The proportionality between realized and predicted evolvabilities tends to fade out for long-term selection responses (\SupMat \ref{supp:r2evolv}), which can be due to the evolution of mutational constraints (the $\M$ matrix evolves compared to the initial network). This was confirmed by tracking the evolution of mutational correlations across robustess traits through time (Figure~\ref{fig:evolcor}). Average correlations did not evolve in the control simulations, but direct selection on robustness components did trigger systematic change in some (but not all) mutational correlations. For instance, the correlation between $\earlymut$ and $\latemut$ does not seem to be evolvable, while the correlation between $\earlymut$ and $\earlyenv$ changed from $\simeq 0.3$ to about $0.6$ or $0.15$ depending on the selection regime. All correlations remained positive. Correlations evolved approximately the same way under univariate and bivariate selection, and the evolution of correlations was driven by the direction of selection (more or less robustness), and not by the orientation of the selection gradient relative to the main evolvability axis. Within each pair of robustness components, the evolution of correlation was consistent: for instance, selecting to decrease $\earlyenv$ or $\earlymut$ (i.e.\ making the network more robust) always decreased the correlation between $\earlyenv$ and $\earlymut$. Nevertheless, there was no general pattern associating the evolution of robustness and the evolution of correlation; for instance, decreasing $\stability$ or $\earlyenv$ increased the correlation between these two variables. 

\begin{figure}[tp!]
\begin{adjustwidth}{-1.5in}{0in}
\begin{flushright}
\includegraphics{results/fig3}
\caption{\color{Gray} \label{fig:evol} The evolution of all five robustness indicators under direct and indirect selection was recorded for 10,000 generations in individual-based simulations. The figures show the average and standard deviation of robustness over 100 replicated simulations. The control simulations (black circles) correspond to stabilizing selection on gene expression only (no direct selection on robustness). Colored symbols correspond to simulations in which robusntess indicators were selected up or down (upward or downward triangles), colors correspond to the observed indicator, colums indicate which indicator was selected (diagnoal panels: direct selection, off-diagonal panels: indirect selection).}
\end{flushright}\end{adjustwidth}
\end{figure}

\begin{figure}[thp!]
\begin{adjustwidth}{-1in}{0in}
\begin{flushright}
\includegraphics{results/fig4}
\caption{\color{Gray} \label{fig:evolvability} Trajectories of the bivariate response to selection over 5000 generations (average over 100 simulation replicates) for all combinations of robustness indicators. Each panel displays the selection response in eight directions (four univariate and four bivariate combinations), as illustrated in the legend. Symbol colors match the same code as Figure~\ref{fig:evol} for univariate responses. Mutational and conditional mutational matrices, estimated from the initial genotypes, are illstrated as ellipses in each panel (95\% ellipses assuming a multivariate Gaussian mutational distribution). For conditional $\bm{\mathrm M}_c$ matrices, the constraining traits were the three gene which expression was under stabilizing selection. X and Y axes were adjusted so that their scale matches for each trait comparison (correlational ellipses were not distorted). The colored inset illustrates the relation between the predicted mutational evolvability (calculated from $\bm{\mathrm M}_c$) and the observed evolvability in the direction of selection after 1000 generations (same color/symbol code as in the rest of the figure, hyphenated line: linear regression with no intercept). }
\end{flushright}\end{adjustwidth}
\end{figure}

\begin{figure}[thp!]
%~ \begin{adjustwidth}{-1in}{0in}
\begin{flushright}
\includegraphics{results/fig5}
\caption{\color{Gray} \label{fig:evolcor} Evolution of the average mutational correlation under univariate and bivariate selection among robustness components. Panels, symbols, and colors are the same as in Figure~\ref{fig:evolvability}. Correlations were estimated for each $\M$ matrix and averaged over 100 simulation replicates. }
\end{flushright}%\end{adjustwidth}
\end{figure}


\section{Discussion}

Whether or not various robustness components of genetic architectures are independent is central to understand why organisms are robust or sensitive to genetic or environmental disturbances. Independent genetic bases of robustness components would call for independent evolutionary histories, while a pleiotropic genetic architecture could explain the evolution of nonadaptive robustness components as a result of indirect selection. The analysis of the genetic correlations between five robustness components, based on a simple gene network model, results in a balanced answer: robustness components are largely correlated, but pleiotropy is not an absolute constraint, and pairs of robustness components evolved in divergent directions under direct bivariate selection. Such a quantitative answer to the so-called 'congruence' hypothesis \citep{dHW+03} would explain both how unselected robustness components could be partly driven by indirect selection and why various robustness-related features seem to have their own evolutionary history. 

\subsection{Model}

Gene regulation networks are popular candidates when attempting to model complex biological processes: they are at least partly built on solid and realistic principles (transcription factors can enhance or repress the expression of other genes), gene regulation plays a crucial role in most biological, physiological, and developmental mechanisms, and even modest size regulation networks display a wide diversity of behavior, including homeostasis (stable equilibrium of gene expressions) \citep{Ste99}, cyclic dynamics \citep{LG03,ARB+10}, or amplification of a weak signal \citep{HB08}. Conveniently, the phenotypic level considered as the output of a gene network (the expression level of all network genes) can be assimilated to a partial transcriptome, which opens the possibility for confrontation with empirical data. 

The gene network model proposed by \citet{Wag94} is particularly popular in evolutionary biology to model gene network evolution due to its computational simplicity and efficiency, combined with a direct biological interpretation (each line of the regulation matrix is the set of transcription factor fixation sites in the promoter of a gene). In practice, multiple variants based on this original model have been derived, either to address specific questions, or to correct for unrealistic features. Here, I used a quantitative version of the model, in which gene expressions were scaled between 0 (no expression) and 1 (maximum expression), which was first proposed in \citet{Wag94}, although later work have often preferred binary networks (in which genes can be on/off, e.g. \citealp{Wag96,CMW07}), and a gene expression scaling between -1 and 1. Unlike in \citet{Wag96, SB02}, mutations had cumulative effects (the value of the mutant allele was drawn in a Gaussian centered around the value of the parental allele), which allows for cumulative evolution. Finally, the sigmoid response function was made asymetrical by introducing a constitutive expression parameter (as in e.g. \citealp{RL16}) in order to avoid the unrealistically high expression of unregulated genes (half the maximum expression) from the default setting. This constitutive expression was not evolvable in the model, but simulations (\SupMat \ref{supp:explo}) show that two robustness components ($\earlyenv$ and $\stability$) were very sensitive to this parameter (larger constitutive expression was associated with more robust networks). It is thus not unlikely that real systems may evolve towards more robustness by increasing the constitutive expression of key genes, as already suggested (for dfferent reasons) by \citet{DW15}. 

Discrete time and simple matrix algebra made it possible to run evolutionary individual-based computer simulations, in which the network output needs to be calculated for thousands of individuals and thousands of generations. Using more realistic models based on continuous time and differential equations, non-linear regulation effects, and independent degradation and transcription rates would make the simulations less practical, with little benefit in terms of explanatory power. Computational constraints also limit the network size to a few dozen genes, which was not enough to generate realistic levels of sparcity -- simulated gene networks were too dense to be realistic. Decreasing network density and smaller network sizes made robustness components slightly less correlated (Figure~\ref{fig:pca}E and~F), suggesting that the integration of robustness components increases with network complexity. The simulated phenotypic target (maintaining a constant set of gene expressions) were also extremely simple compared to what gene networks are theoretically able to do (e.g. converging to different equilibria in different cell types, or controling a complex dynamic of gene expression during the development). However, the simulation results are robust to most simulation parameters (\SupMat \ref{supp:explo}), suggesting that they reflect general properties of the underlying genetic architecture. 

In spite of the simplicity of the network model, it appeared that connecting network features (for instance, the strength of a specific regulation) and robustness was not trivial, even in very small networks. For instance, in the $n=2$ gene-network analysis, most robustness components were complex functions of all four regulation strengths. Throughout this work, robustness was thus treated as an emergent property of the underlying network, which can not be easily deduced from a reductionnist approach. Yet, it is possible to interpret the correlation patterns in terms of network dynamics. Two of the most correlated components are the robustness to early environmental variation $\earlyenv$ and network stability $\stability$, which both measure the ability of the network to converge to a given gene expression equilibrium. Conversely, the correlation between late mutational $\latemut$ and environmental $\lateenv$ robustnesses can be attributed to the consequences of such disturbances over a single time step: for a single target gene, decreasing the concentration of a transcription factor and decreasing the sensitivity of the promoter to the same transcription factor have very similar immediate consequences on gene expression. Yet, even if these measurements happen to be correlated by construction in the network model, their partial evolutionary independence highlights their potential for independent evolvability in real gene networks, which are substantially more complex and subtle than our mathematical approximation. 

\subsection{Measuring robustness}

There are potentially many ways to measure the robustness of a phenotypic trait. Here, five indicators were proposed to catch various (and potentially independent) aspects of what is generally defined as robustness. The sensitivity to inherited mutations ($\earlymut$) is probably the most popular one, as it is central to the discussion around the evolution of canalization \citep{Wad59, Wag96, Far15}. The sensitivity to environmental perturbations is also unavoidable, although its implementation in a gene network model is less straightforward. Here, it was calculated as both the sensitivity of the network to disturbance in the initial expression state ($\earlyenv$), which measures the size of the basin of attraction of the optimal expression pattern, and as the strength of the stability of the equilibrium when disturbed ($\lateenv$). These two measurements can be interpreted as developmental robustness and physiological homeostasis, respectively, as they quantify the response of the network to disturbances in the expression levels at different time scales. The robustness to mutations occuring after the network convergence ($\latemut$) was considered because it sets up an alternative to the genetic vs.\ environmental congruence hypothesis: in long-lived organisms, non-heritable (somatic) mutations participate to the ageing process \citep{KLH12}, ageing being to some extent under direct selection. Thus, the robustness to somatic mutations could also drive indirectly the evolution of genetic canalization. Althought not strictly a robustness component, the gene network stability ($\stability$, amplitude of the fluctuations of gene expressions) was also considered because it has been proven to drive an indirect response of genetic canalization, based on very similar model \citep{SB02}. Its correlation with other robustness indicators confirms the tigh link between robustness and stability in gene networks.

These indicators were chosen based on the possibility to measure them in numerical simulations. Although the empirical assessment of the correlation between robustness components would be way more convincing than a theoretical study, defining similar measurements from experimental datasets can be challenging. For instance, $\earlymut$ and $\earlyenv$ could, at least in theory, be estimated as the variance in gene expression across genetic backgrounds or across environmental conditions, respectively. Measuring $\latemut$ environmentally is more complicated, as it would likely be confounded with other ageing mechanisms. In contrast, the empirical distinction between e.g.\ $\lateenv$ and $\stability$ relies on discriminating internal vs.\ external sources of noise, and might be in practice impossible. In all cases, gene expression data are generally quite noisy and their analysis necessitates heavy corrections to prevent multiple testing issues. Studying empirically the robustness and evolvability of molecular and morphological traits has long been considered as a challenging task, but methodological and technological progress has recently brought new concrete perspectives \citep{PW19}. 

Some popular measurements of developmental robustness were not considered here for technical reasons. For instance, fluctuating asymetry (the variance between the same phenotypic trait measured in the right and the left body parts of symmetric organisms) is a convenient measurement of microenvironmental effects on the development \citep{DD01,LK05}, but it has no equivalent at the level of gene expression in a regulation network. The deterministic sensitivity to a directional environmental gradient could also be used to measure phenotypic plasticity, which is central to the question of phenotypic robustness. Yet, there are several ways to model phenotypic plasticity in a gene network \citep{Mas04,BTL21}, and it requires a specific selection setup (different expression optimums as a function of the environment). Because of this additional complexity, adaptive phenotypic plasticity was excluded from the focus of this work, although the evolution of plasticity of gene expression remains an intriguing and fundamental question. In particular, phenotypic plasticity (i.e. an adaptive lack of robustness to some environmental signal) may itself be canalized to genetic or other environmental disturbances \citep{SK94}; considering reaction norms (a measurement of plasticity) as quantitative traits thus opens challenging questions about the adaptive evolution of the canalization of robustness traits.

\section*{Acknowledgements}

Many thanks to Laurent Loison for insightful discussions. Simulations were partly performed on the Core Cluster of the Institut Français de Bioinformatique (IFB) (ANR-11-INBS-0013) .

\section*{Conflict of interest disclosure}

The author of this article declares that he has no financial conflict of interest with the content of this article.

\printbibliography

\clearpage

\begin{appendices}
  \setcounter{section}{0}
  \renewcommand{\thesection}{\arabic{section}}

  \section{}
    \label{supp:sensitmag}
    \subsection*{Sensitivity of the robustness measurements to the magnitude of the disturbance}
	\begin{center}
	\includegraphics{results/figS1}
	\end{center}
	{\color{Gray} Four out of five robustness indicators ($\earlyenv$, $\lateenv$, $\earlymut$, $\latemut$) depend on the magnitude of the disturbance ($\sigma_E$, $\sigma_e$, $\sigma_M$, and $\sigma_m$, respectively). The figure displays the influence of the size of the disturbance on the robustness measurement (left: 10 random networks, right: 10 evolved networks). Vertical dotted lines stand for the values used in the simulations. Robustness scores are not completely consistent for random networks, as some of them can be differentially robust to large or small disturbances. The consistency is better in evolved networks (the rank of different genotypes in terms of robustness rarely depends on the size of the disturbance).  }


  \clearpage
  \section{}
    \label{supp:fullcorr}
    \subsection*{Correlations among robustness indexes among random networks}
	\begin{adjustwidth}{-1in}{0in}
	\begin{flushright}
	\includegraphics{results/figS2}
	\end{flushright}

	\color{Gray} Correlations between all five robustness components among 10,000 random 6-gene networks ($\mu_0=0, \sigma_0=1$). 
	\end{adjustwidth}


  \clearpage
  \section{}
    \label{supp:PC}
    \subsection*{Sampling effects on Principal Components}
    
	\begin{center}
	\includegraphics{results/figS3}
	\end{center}    

	{\color{Gray} Influence of the sampling effect (number of networks and number of replicates $R$ to estimate robustness) on the relative
	weight of the principal components. All PCs except the last one are robust to sampling. }


  \clearpage
  \section{}
    \label{supp:whyitfails}
    \subsection*{Reasons for not reaching the desired equilibrium}
	\begin{center}
	\includegraphics{results/figS4}
	\end{center}
	
	{\color{Gray} Although equation~\ref{eq:predef} guarantees that an equilibrium exists at the target phenotypic expression, the equilibrium might not be reachable in practice when simulating the gene network dynamics. The colored area in the figure corresponds to networks that failed to produced the target phenotype, each color representing a distinct reason; Yellow: network dynamics was slow and the final gene expression has not been reached yet after 16 time steps; Gray: an alternative equilibrium was reached (most of the time implying that one or both genes are either completely silenced to fully expressed). Red: The network steady state featured oscillations that were so large that they hit the maximum or minimum expression, shifting the average expression away from the target expression. }
	
	
  \clearpage
  \section{}
    \label{supp:W}
    \subsection*{Two-gene example networks}

	\begin{center}
	\begin{tabular}{rrrrr}
	  \hline
	 & $W_{11}$ & $W_{21}$ & $W_{12}$ & $W_{22}$ \\ 
	  \hline
	  A & 0.70 & 0.20 & -0.21 & 0.38 \\ 
	  B & -0.30 & 0.30 & 0.29 & 0.33 \\ 
	  C & -0.40 & 0.80 & 0.34 & 0.08 \\ 
	  D & -1.00 & -0.80 & 0.64 & 0.88 \\ 
	  E & 1.50 & 3.50 & -0.61 & -1.27 \\ 
	   \hline
	\end{tabular}
	\end{center}
	{\color{Gray} The five two-gene networks detailed in Figure~\ref{fig:imgpanels} and \SupMat~\ref{supp:simpanels}.}

	
  \clearpage
  \section{}
    \label{supp:simpanels}
    \subsection*{Illustration of the robustness scores}

	\begin{adjustwidth}{-1in}{0in}
	\includegraphics{results/figS6}

	{\color{Gray} The figure displays a subset of the replicated tests for four robustness indexes. Rows A to E correspond to the five networks described in \SupMat \ref{supp:W}. In each panel, the default (undisturbed) network kinetics is displayed as plain lines (black for gene~1, red for gene~2), while 10 disturbed networks are indicated as pale lines. By construction, all networks have an equilibrium at (0.3, 0.6). The network robustness to genetic disturbance was estimated by mutating the gene network before the first time step (early genetic mutation, first column) or before the last time step (late genetic mutation, second column). Environmental robustness was estimated by disturbing the gene expression, without changing the genotype, before the first time step (early environmental, third column) and before the last time step (late environmental, fourth column). The network stability can be assessed from the amplitude of the cycles in the undisturbed kinetics (e.g.\ in network C), and does not rely on a stochastic algorithm.}
	\end{adjustwidth}


  \clearpage
  \section{}
    \label{supp:explo}
    \subsection*{Exploration of the parameter set}
	\begin{adjustwidth}{-2.5in}{0in}
	\includegraphics{results/figS7} 
	
	{\color{Gray}Influence of simulation parameters (mutation rate $\nu$, mutation size $\sigma_\nu$, population size $N$, constitutive expression $a$, total number of genes $n$, number of selected genes $n^\prime$, network density $d$, and strenght of selection $s$) on fitness and robustness indexes after 5000 generations (default settings except for the target parameter). The figure reports the mean $\pm$ standard deviation across 20 replicated simulations. Vertical dotted lines stand for the default parameter values. }
	\end{adjustwidth}


	
  \clearpage
  \section{}
    \label{supp:r2evolv}
    \subsection*{Accuracy of the prediction vs.\ simulation time}
	\begin{center}
	\includegraphics{results/figS8} 
	\end{center}
	
	{\color{Gray} Effect of the number of generations on the proportionality relationship between predicted and observed evolvabilities of robustness components. The figure displays the $r^2$ of a linear regression (without intercept) between the predicted evolvability from the conditional $\M_c$ matrix measured at the first generation and the observed evolvability in the direction of selection for all replicated simulations. The regression at generation 1,000 is illustrated in the colored inset in Figure~\ref{fig:evolvability}. }
	
\end{appendices}

\end{document}
