\documentclass[a4paper]{article}
\usepackage[utf8]{inputenc}

\usepackage[margin=3cm]{geometry}

\usepackage{setspace}
\doublespacing

\usepackage{biblatex}
\bibliography{robustness.bib}

\usepackage{bm}
\usepackage{amsmath}

\usepackage{graphicx}

\usepackage[english]{babel}

\usepackage[page,title]{appendix}
\addto\captionsenglish{%
	\renewcommand{\appendixname}{Supplementary material}
	\renewcommand{\appendixpagename}{}%Supplementary material}
}


\newcommand{\stability}{{\rho_S}}
\newcommand{\earlyenv}{{\rho_E}}
\newcommand{\lateenv}{{\rho_e}}
\newcommand{\earlymut}{{\rho_M}}
\newcommand{\latemut}{{\rho_m}}

\newcommand{\SupMat}{Supplementary material~}

\title{Gene network robustness as a multivariate character}
% Robustness as a multivariate character

\author{Arnaud LE ROUZIC}
\date{}

\begin{document}

\maketitle

\section{Introduction}

% Robustness: definition, history of the concept. 
Robustness is the capacity of living organisms to buffer internal or environmental disturbances. Robustness encompass, for instance, the ability to maintain physiological equilibria (homeostasis), to ensure developmental stability, or to repair and mitigate DNA damage in both soma and germline. Although robustness is virtually intermingled with the definition of life itself, its underlying mechanisms and its evolutionary origins remain far from being clearly understood. 

% Robustness in evolutionary biology: canalization, pop gen, state of the art. 
The evolutionary ... roots into the conceptual and empirical work of C.H. Waddington and the notion of canalization (Wadd, Small, + recent review?). Canalization is a property of complex developmental systems that buffers environmental and genetic variation, and maintains actively the organism in an optimal developmental path. Although the scope and the definition of canalization varies substantially among authors, canalization is generally expected to evolve as an adaptation to "canalizing" selection for an optimal phenotype. However, formal population genetic models have questioned the unicity of the canalization process. In particular, robustness to environmental factors appears more likely to evolve as an adaptation than robustness to genetic (mutational) disturbances, on which selection seems to be rather weak and indirect even in optimal theoretical conditions (Wagner et al, Le Rouzic et al). In contrast, evolution of environmental sensitivity could evolve as a direct response (phenotypic plasticity), while evolution of "evolvability" remains dubious. 

% Congruence
In this context, the evolution of robustness as a general property of the organisms heavily depends on the integration of the different robustness dimensions. If the robustness to environmental factors and to genetic mutations share the same physiological bases, the adaptive evolution of environmental canalization could generate a correlated response of genetic canalization, which could thus exist without a specific evolutionary mechanism: this is the "congruence" hypothesis, as defined in deVisser 2003. In contrast, if genetic and environmental robustness were independent biological mechanisms, they would be featured by independent evolutionary mechanisms, and possibly independent evolutionary histories. 

Although this issue would benefit from a better theoretical framework, modelling the evolution of robustness is not straightforward. The simplest approach relies on focusing on modifiers, i.e. genes that can influence the robustness of the organism without affecting the phenotype. However, in the case of robustness, modifier-based models are either restricted to very simple cases, rely on tricky rescaling, or fail at dissociating the phenotype and the robustness to the phenotype (Wagner et al, Kawecki...). In addition, in such models, any correlation between e.g. environmental and genetic robustness would appear as a modelling choice, and not an output of the model. More promising are models in which the phenotype is a result of an integrated process mimicking some developmental or regulation processes. In such dynamic models, robustness to various disturbances appear as an emergent property of the regulation feedback mechanisms that cannot be easily deduced from the model parameters. Although the potential palette of relevant dynamic models is large and could include e.g. morphological development models (Salazar-Ciudad?) or metabolic models (Liver model?), evolutionary biologists have often considered gene regulatory network models, inspired from Wagner 1996, as a good compromise between complexity and numerical tractability for studying the evolution of canalization and robustness. 

Robustness to mutations: Wagner 1996, Runneburger \& Le Rouzic 2006. Robustness to environment \& Canalization: Masel. Stability of gene expression: Siegal \& Bergman 2002. 

Here, I aim at extending this approach to address specifically the multidimensional nature of robustness components in gene networks, and to study their evolutionary independence. Five robustness indicators were considered, two of them corresponding to environmental robustness (early vs.\ late disturbances), two corresponding to genetic robustness (early --- inherited --- or late --- acquired --- mutations), the last one (gene expression unstability) being related to the intrinsic stability of the expression phenotype. The results show that although all robustness indicators tend to be correlated, they are not absolutely constrained and could diverge under suitable selection pressures. 

\section{Model and Methods}

\subsection{Gene regulatory network}

The network model is derived from \cite{Wag94,Wagner96} and belongs to the family of GRN sometimes refered to as "Wagner model" (see \cite{FierstPhuillips2016} for the historical record). The main difference with the original Wagner model is that the network output (gene expressions) is quantitative and not qualitative, in the same way as in \cite{SB02}. 

More specifically, the structure of a $n$-gene network is encoded as a $n\times n$ matrix $\bm W$, while the state of the network is stored into a vector of size $n$, $\bm S$. In this setting, $W_{ij}$ represents the influence of gene $j$ on the expression of gene $i$, $W_{ij} < 0$ represents a negative interaction (inhibition), $W_{ij} > 0$ a positive interaction (activation), and $W_{ij} = 0$ denotes the absence of regulatory interaction. $S_i$ is the expression of gene $i$, ranging between 0 (no expression) and 1 (maximum expression). 

The properties of these gene networks are explored in a discrete dynamic system:
\begin{equation}
Â \bm P_{t+1} = F(\bm W \bm P_t),
\end{equation}
\noindent where the function $F$ is a vectorized version of a sigmoid scaling function: $F(x_1, x_2, \dots, x_n) = [f(x_1), f(x_2), \dots, f(x_n)]$ with
\begin{equation} \label{eq:fx}
f(x) = \frac{1}{1+ \lambda_a e ^{- \mu_a x}}, 
\end{equation}
\noindent with $\lambda_a = (1-a)/a$ and $\mu_a = 1/a(1-a)$. The function $f$ is scaled such that $f(0) = a$ and $df/dx|_{x=0}=1$; the parameter $a$ thus stands for the constitutive gene expression (the expression of a gene in absence of regulators), and this function defines the scale of the matrix $\bm W$, as a small effect $W_{ij} = \delta$ on an otherwise non-regulated gene leads to the same change in gene expression $S_i = a+\delta$. 
%Setting $a=0.5$ leads to $f(x) = 1/(1+e^{-\mu x})$ with $\mu = 4$, which can be directly compared with e.g.\ \cite{PBF12}. 

Gene networks dynamics start from an initial expression $S_0$, and gene expression was updated for $T$ timesteps. By default, $P_0 = (a, a, ..., a)$, since this step immediately follows a virtual initial state with no expression. The expression phenotype corresponding to a gene network was determined by averaging gene expressions during the last $\tau$ timesteps for each gene $i$: $\bar P_i = \sum_{t=T-\tau}^T P_{it}$. 

\subsection{Robustness indicators}

Five robustness indicators were calculated, corresponding to five different aspects of genetic or environmental robustness in a gene network: network stability $\stability$, robustess to early ($\earlyenv$) and late ($\lateenv$) environmental disturbance, and robustness to early ($\earlymut$) and late ($\latemut$) genetic disturbance. All indicators were expressed on a scale homogeneous to log variances in gene expressions.

Dynamic systems based on the Wagner model often tend to generate limit cycles and never converge to a stable equilibrium. Network stability $\stability$ quantifies the capacity for a specific network to lead to stable gene expressions.  For consistency with other indicators, this unstability was measured as the log squared difference between the average expression during the last $\tau$ timesteps, and an extra timestep: $\stability_i = \log( \bar P_i - P_{T+1})^2)$. 

The robustness to early environmental disturbance $\earlyenv$ measures the capacity of a network to reach a consistent final state starting from different initial gene expressions. In practice, $R$ replicates of the network dynamics were run, in which the initial gene expressions ($P_0$) were drawn into Gaussian ($\mu = a, \sigma = \sigma_e$) distributions. The environmental robustness for each gene $i$ was measured as the log variance in the final gene expression across these replicates: $\earlyenv_i = \log [ 1/(R-1) \sum_{r=1}^R (\bar P_{i,r} - (1/R) \sum _r \bar P_{i,r})^2 ]$. 

The robustness to late environmental disturbance $\lateenv$ measures the capacity of a network to recover its equilibrium state after having being disturbed. Gene expressions after $T$ timesteps were disturbed by adding a random Gaussian noise of standard deviation $\sigma_E$ to each gene of the network, and $\lateenv_i$ was computed for each gene $i$ as the log variance in gene expression at time step $T+1$ over $R$ replicates: $\lateenv_i = \log [ 1/(R-1) \sum_{r=1}^R (P_{i,T+1,r} - (1/R) \sum_r P_{i,T+1,r})^2)]$. 

The robustness to early mutations $\earlymut$ measures the system robustness to inherited genetic mutations (modifications of the $\bm W$ matrix). A random non-zero element of the $\bm W$ matrix was shifted by a random Gaussian number of standard deviation $\sigma_m$, and its consequences on the mean expression of all network genes was recorded. The procedure was replicated $R$ times, and the robustness score $\earlymut_i$ for each gene $i$ was calculated as the log variance of gene expression across $R$ replicates: $\earlymut_i = \log[ 1/(R-1) \sum_{r=1}^R (\bar P_{i,r} - (1/R)\sum_r \bar P_{i,r})^2]$. 

Finally, the robustess to late mutations $\latemut$ measured the effect of mutations in the gene network $\bm W$ after having reached the final state. In practice, the $\bm W$ matrix was mutated in the same way as for $\earlymut$, but its consequences on gene expression were calculated for only one timestep, starting from the final state of the network. The procedure was simulated $R$ times, $\latemut_i = \log[ 1/(R-1) \sum_{r=1}^R (P_{i,T+1,r} - (1/R) \sum_r P_{i,T+1,r})^2 ]$.

All these scores were calculated for every gene $i$ of a given network, and then averaged over all genes in order to get a series of summary network descriptors 

\subsection{Random networks}

Random networks were generated as $n\times n$ matrices filled with independent identically-distributed random numbers drawn into a Gaussian ($\mu_0, \sigma_0$) distribution. The parameters of the distribution were determined so that gene expressions $P_i$ was as uniformy distributed as possible between minimum ($P=0$) and maximum ($P=1$) expressions; this corresponded to $\mu_0=-0.2$ and $\sigma_0=1.2$ in a full-density 6-gene networks, A density parameter $1/n \leq d \leq 1$ could be specified, corresponding to the frequency of non-zero slots in the $\bm W$ matrix. Zeros were placed randomly, with the constraint that all genes should be regulated by at least another one. 

\subsection{Exhaustive exploration of 2-gene networks}

The main interest of gene-network models is the complexity and the richness of the underlying genotype-phenotype. As a side effect, such models are in general difficult to handle mathematically (\cite{CTH11,LP12}). Excluding the one-gene self-regulating case (which already has non-trivial mathematical properties), the simplest network (2-by-2 matrix) has four genetic parameters, which makes the exploration of the parameter set tedious. Here, the number of dimensions was restricted by considering the set of networks that lead to a predefined arbitrary equilibrium, $\bm P^\ast_\infty = (P^\ast_1, P^\ast_2)$. As $F(\bm W \bm P^\ast_\infty) = \bm P^\ast_\infty$, the $\bm W$ matrix can be reduced to two independent parameters, $W_{11}$ and $W_{21}$:
\begin{equation}
    \bm W = F \left [\begin{pmatrix} W_{11} & A \\ W_{21} & B \end{pmatrix} Â \begin{pmatrix} P^\ast_1 \\ P^\ast_2 \end{pmatrix} \right] = \begin{pmatrix}P^\ast_1 \\ P^\ast_2 \end{pmatrix},
\end{equation}
\noindent with
\begin{equation}
    \begin{split}
        A = \frac{1}{P^\ast_2} [f^{-1}(P^\ast_1)-W_{11}P^\ast_1], \\
        B = \frac{1}{P^\ast_2} [f^{-1}(P^\ast_2) - W_{21} P^\ast_1],
    \end{split}
\end{equation}
\noindent $f^{-1}(x) = -\frac{1}{\mu_a} \log \left( \frac{1-x}{\lambda_a x} \right)$ being the inverse of $f(x)$ (equation~\ref{eq:fx}). The $\bm W$ matrix achieving the desired $\bm P^\ast_\infty$ equilibrium from a specific pair $W_{11}, W_{21}$ always exists (and is unique), but the stability of the equilibrium is not guaranteed. Networks which final gene expression $\bm P = (P_1,P_2)$ differed substantially from the target (in practice, when $|P_1 - P^\ast_1|Â + |P_2 - P^\ast_2| > 0.15$) were excluded from the analysis. Such discrepancies correspond to either unstable equilibria (in which case gene expressions were driven away from the equilibrium until reaching an extreme (0 or 1) expression) or extreme oscillatory behaviors (large oscillations may hit extreme expression limits (0 or 1) which drives the average expression away from the target equilibrium). 

\subsection{Evolutionary simulations}

The evolution of gene networks under various evolutionary constraints was studied by individual-based simulations. Each individual was featured by its genotype (a $n \times n$ $\bm W$ matrix), its expression phenotype $\bar P$, and the five robustness scores $\stability$, $\earlyenv$, $\lateenv$, $\earlymut$, and $\latemut$. For simplicity, individuals were haploid and reproduced clonally. Mutations consisted in adding a random Gaussian deviate of variance $\sigma_m^2$ to an element of the $\bm W$ matrix, with a rate $\mu$. Generations were non-overlapping, and population size $N$ was constant. A generation consisted in sampling $N$ new individuals among the $N$ parents, with a probability propotional to the individual fitness. Fitness was computed assuming stabilizing selection around a target (optimal) expression level for $n^\prime \leq n$ genes of the network, as $w = \exp(- \sum_{i=1}^{n^\prime} s_i (\bar P_i - \theta_i)^2 )$, where $s_i$ was the strength of stabiizing selection on gene $i$ ($s_i = 0$ standing for no selection), and $\theta_i$ was the optimal expression phenotype. In most cases, the $\theta_i$ were drawn in uniform (0,1) distributions at the beginning of each replicated simulation. 

In addition, direct, directional selection on robustness indicators was performed in some simulations, consisting in multiplying individual fitness by $\exp ( \sum_{x \in (S, E, e, M, m)} \beta_x \rho _x )$, where $\beta_x$ is the strength of directional (positive or negative) selection on robustness index $x$. 

Simulations and data analysis were coded in R \citep{R20}, except for the gene network dynamics that was coded in C++ and embedded in the R code with the Rcpp package \citep{EF11}. 

\section{Results}

\subsection{Random networks}

Random interaction matrices are regularly used in the literature to study the general properties of gene networks (e.g.\ \cite{CTH11,PBF12}). As such, random networks are not expected to reflect the properties of biologically-realistic genetic architectures, as biological networks are far from random. However, such an approach helps developing a general intuition about the properties of the underlying model, especially about the amount of selection it takes to bring a network into its non-random state. 

Correlations were calculated between all five robustness components over 10,000 random networks (\SupMat~\ref{supp:fullcorr}). All robustness components were positively correlated. However, correlations range from above 0.9 (early vs.\ late mutations) to about 0.2 (stability vs early mutations). A Principal Component Analysis (Figure~\ref{pca}) confirms that robustness components are not tighly associated. The first PC (57\% of the total variance) corresponds to the general robustness of the network, and involves all robustness indexes. The remaining variance is explained by orthogonal vectors separating all other robustness components. Although weak (about 2\% of the total variance), the last PC still represents an independent direction of variation and is not an artifact of the sampling effect (\SupMat~\ref{supp:samplingPC}). 

\begin{figure}[t]
\begin{center}
\includegraphics[width=10cm]{../results/figB}
\caption{\label{fig:pca} Summary of the principal component analysis on the five robustness components over 10,000 random 6-gene networks ($\mu_0=-0.2, \sigma_0=1.2$). Left: position of the five robustness components on all five (normalized) Principal Components (PC); $\stability$: Stability, $\earlyenv$: Early environmental, $\lateenv$: Late environmental, $\earlymut$: Early genetic, $\latemut$: Late genetic. Right: relative contribution of the five PCs to the total variance.}
\end{center}
\end{figure}

\subsection{2-gene networks}

In the following, I considered the particular case of $\bm S_\infty$ = (0.3, 0.6) with two intermediate, distinct gene expressions. Equivalent results could be achieved with a different target. Figure~\ref{fig:imgpanels} illustrates how the robustness components vary in this constrained 2-gene network model (red stands for maximum robustness, i.e.\ minimum scores for $U$,$H$,$E$,$C$, and $K$). In the white regions, the equilibrium was not achieved in numerical simulations for distinct reasons. In the bottom-right region, the target $\S_\infty$ equilibrium was unstable, and the model converged to another equilibrium if the starting point was different from $\bm S_\infty$. In the top-left region, fluctuations around the equilibrium were so large that they hit the edges of the (0,1) gene expression interval, shifting the average gene expression towards 0.5. 

\begin{figure}[t]
\begin{center}
\includegraphics[width=15cm]{../results/figC}
\caption{\label{fig:imgpanels} Robustness indicators ($\stability$, $\earlyenv$, $\lateenv$, $\earlymut$, $\latemut$) estimated for an exhaustive continuum of two-gene networks with an arbitrary expression equilibrium at $\bm P_\infty = (0.3, 0.6)$. Although two-gene networks have four independent genetic parameters, only two are represented here, the two others being computed to ensure the desired equilibrium. Red stands for the maximum robustness (lowest scores); yellow for minimum robustness (highest scores). For readability, color scales are different across panels. Letters A to E stand for five example networks illustrated in \Supmat~\ref{supp:simpanels}.}
\end{center}
\end{figure}

It immediately appears that the different robustness components may be somehow correlated, but do not overlap. The simulated behavior of networks A to E are illustrated in \SupMat~\ref{supp:simpanels}, and the corresponding $\bm W$ matrices are provided in \SupMat~\ref{supp:W}. The network denoted as "B" is robust to all five components, while network E is sensitive to all components except stability. Network C is unstable, but remains relatively canalized. Networks A and D illustrate intermediate decanalization behaviors, through different mechanisms (unstability for network D, and weak buffering for network A). 

This 2-gene network analysis thus confirms the results obtained for larger gene networks. Robustness is thus not a feature of large and intricate genetic architectures, as it is already present (and multidimensional) in the simplest gene networks. The different robustness components are partially independent, even in small and highly constrained networks. All the networks considered here converge to the same gene expression at time step 20, and can thus be considered as phenotypically equivalent ; the colored space in Figure~\ref{fig:imgpanels} thus represents a connected neutral network in which populations could evolve, and thus change the topology and the robustness of the gene network, while keeping the expression phenotype constant. 

\subsection{Network evolution}

\subsection{Genetic and Mutational covariances}




\section{Discussion}

\printbibliography




\clearpage



\begin{appendices}
  \setcounter{section}{0}
  \renewcommand{\thesection}{\arabic{section}}

  \clearpage
  \section{}
    \label{supp:fullcorr}
    \subsection*{Correlations among robustness indexes}

	\includegraphics[width=15cm]{../results/figA}
	
	Correlations between all five robustness components among 10,000 random 10-gene networks ($\mu_0=-0.2, \sigma_0=1.2$). 



  \clearpage
  \section{}
    \label{supp:W}
    \subsection*{Two-gene example networks}

	\begin{center}
	\begin{tabular}{rrrrr}
	  \hline
	 & $W_{11}$ & $W_{21}$ & $W_{12}$ & $W_{22}$ \\ 
	  \hline
	  A & 0.70 & 0.20 & -0.21 & 0.38 \\ 
	  B & -0.30 & 0.30 & 0.29 & 0.33 \\ 
	  C & -0.40 & 0.80 & 0.34 & 0.08 \\ 
	  D & -1.00 & -0.80 & 0.64 & 0.88 \\ 
	  E & 1.50 & 3.50 & -0.61 & -1.27 \\ 
	   \hline
	\end{tabular}
	\end{center}
	
	The five two-gene networks detailed in figures~\ref{fig:imgpanels} and~\ref{fig:simpanels}.


  \clearpage
  \section{}
    \label{supp:simpanels}
    \subsection*{Illustration of the robustness scores}

	\includegraphics[width=15cm]{../results/figD}
	
	The figure displays a subset of the replicated tests for four robustness indexes. Columns A to E correspond to the five networks described in \SupMat~\ref{supp:W}. In each panel, the default (undisturbed) network kinetics is displayed as plain lines (black for gene~1, red for gene~2). By construction, all networks have an equilibrium at (0.3, 0.6). The network stability can be assessed from the amplitude of the cycles in the undisturbed kinetics. Network homeostasis (first line) was estimated by disturbing the expression state of the genes (for readability, the network response was expanded over several time steps). Environmental canalization (second line) was estimated by disturbing the initial state (semi-transparent lines). Genetic canalization (third line) was measured as the variance in the final state when mutating the network at the beginning of the development. Finally, the robustness to late mutations (last line) consisted in mutating the network after having reached the equilibrium.

\end{appendices}

\end{document}
