\documentclass[10pt,a4paper]{article}
\usepackage[utf8]{inputenc}


%~ \usepackage{cite}
%~ \makeatletter
%~ \renewcommand{\@biblabel}[1]{\quad#1.}
%~ \makeatother


\usepackage{nameref,hyperref}
\usepackage[right]{lineno}

\usepackage[top=0.85in,left=2.75in,footskip=0.75in,marginparwidth=2in]{geometry}

% headrule, footrule and page numbers
\usepackage{lastpage,fancyhdr,graphicx}
\usepackage{epstopdf}
\pagestyle{myheadings}
\pagestyle{fancy}
\fancyhf{}
\rfoot{\thepage/\pageref{LastPage}}
\renewcommand{\footrule}{\hrule height 2pt \vspace{2mm}}
\fancyheadoffset[L]{2.25in}
\fancyfootoffset[L]{2.25in}

\raggedright
\setlength{\parindent}{0.5cm}
\textwidth 5.0in 
\textheight 8.75in

\usepackage{changepage}

\usepackage{microtype}
\DisableLigatures[f]{encoding = *, family = * }

\usepackage{color}
\definecolor{Gray}{gray}{.25}


\usepackage{graphicx}
\usepackage{sidecap}
\usepackage{wrapfig}
\usepackage[pscoord]{eso-pic}
\usepackage[fulladjust]{marginnote}
\reversemarginpar

\usepackage[natbib=true,style=alphabetic,citestyle=authoryear]{biblatex}
\bibliography{robustness.bib}

\usepackage{bm}
\usepackage{amsmath}

\usepackage[english]{babel}

\usepackage[page,title]{appendix}
\addto\captionsenglish{%
	\renewcommand{\appendixname}{Supplementary material}
	\renewcommand{\appendixpagename}{}%Supplementary material}
}


% Replacements in the text

\newcommand{\stability}{{\rho_S}}
\newcommand{\earlyenv}{{\rho_E}}
\newcommand{\lateenv}{{\rho_e}}
\newcommand{\earlymut}{{\rho_M}}
\newcommand{\latemut}{{\rho_m}}

\newcommand{\SupMat}{Supplementary material~}

\title{}
\date{}

\begin{document}
\vspace*{0.35in}

% title goes here:
\begin{flushleft}
{\Large
\textbf\newline{Gene network robustness as a multivariate character}
}
\newline
% authors go here:
\\
Arnaud Le~Rouzic\textsuperscript{1,*}
\\
\bigskip
\bf{1} Laboratoire Évolution: Génomes, Comportement, Écologie; Université Paris-Saclay, CNRS, IRD.
\\
\bigskip
* arnaud.le-rouzic@egce.cnrs-gif.fr

\end{flushleft}

\section*{Abstract}

Robustness to genetic or environmental disturbances is often considered as a key property of living systems. Yet, despite being discussed since about 70 years, how robustness emerges from the complexity of genetic architectures and how it evolves still remain largely misunderstood. In particular, whether robustness to various sources of perturbations at different developmental stages is genetically independent conditions the possibility to propose adaptive scenarios. For instance, selection for the robustness to heritable mutations is known to be modest, and the evolution of genetic canalization might result from indirect selection on a pleiotropically-related character(e.g.\ homeostasis) rather than adaptation. Here, I propose to consider various robustness measurements as quantitative characters, and study theoretically their propensity to evolve independently. Based on a simple evolutionary model of a gene regulatory network, I showed that different ways to measure the robustness of gene expression patterns to genetic or non-genetic disturbances were substantially correlated. Yet, genetic variation for robustness could be detected in at least four dimensions, and robustness components could evolve differentially under direct selection pressure. Therefore, the fact that the sensitivity of gene expression to e.g. mutations and environmental factors rely on the same gene network does not preclude that robustness components may have distinct evolutionary histories. 


\linenumbers

\section{Introduction}

% Robustness: definition, history of the concept. 
Robustness is the capacity of living organisms to buffer internal or environmental disturbances. Robustness encompass, for instance, the ability to maintain physiological equilibria (homeostasis), to ensure developmental stability, or to repair and mitigate DNA damage in both soma and germline. Although robustness is virtually intermingled with the definition of life itself, its underlying mechanisms and its evolutionary origins remain far from being clearly understood \citep{HGK+19}. 

% Robustness in evolutionary biology: canalization, pop gen, state of the art. 
The study of the evolutionary mechanisms leading to robustness roots into the conceptual and empirical work by C.H. Waddington and the notion of canalization (Wadd, Small, + recent review?). Canalization is a property of complex developmental systems that buffers environmental and genetic variation, and maintains actively the organism in an optimal developmental path. Although the scope and the definition of canalization varies substantially among authors, canalization is generally expected to evolve as an adaptation to "canalizing" selection for an optimal phenotype \citep{DD01}. However, formal population genetic models have questioned the unicity of the canalization process. In particular, robustness to environmental factors appears more likely to evolve as an adaptation than robustness to genetic (mutational) disturbances, on which selection seems to be rather weak and indirect even in optimal theoretical conditions (Wagner et al, Le Rouzic et al). In contrast, evolution of environmental sensitivity could evolve as a direct response (phenotypic plasticity), while evolution of "evolvability" remains dubious. 

% Congruence
In this context, the evolution of robustness as a general property of the organisms heavily depends on the integration of the different robustness dimensions. If the robustness to environmental factors and to genetic mutations share the same physiological bases, the adaptive evolution of environmental canalization could generate a correlated response of genetic canalization, which could thus exist without a specific evolutionary mechanism: this is the "congruence" hypothesis, as proposed in \citep{dHW+03}. In contrast, if genetic and environmental robustness were independent biological mechanisms, they would be featured by independent evolutionary mechanisms, and possibly independent evolutionary histories. 

Although this issue would benefit from a better theoretical framework, modelling the evolution of robustness is not straightforward. The simplest approach relies on focusing on modifiers, i.e. genes that can influence the robustness of the organism without affecting the phenotype. However, in the case of robustness, modifier-based models are either restricted to very simple cases, rely on tricky rescaling, or fail at dissociating the phenotype and the robustness to the phenotype \citep{WBB97, Kaw00}. In addition, in models where the genotype-phenotype association is arbitrary, any correlation between e.g.\ environmental and genetic robustness would appear as a modelling choice, and not an output of the model. More promising to address the congruence issue are models in which the phenotype is a result of an integrated process mimicking some developmental or physiological mechanism. In such dynamic models, robustness to various disturbances appear as an emergent property of the model complexity, caused by regulatory feedbacks, that cannot be easily deduced from the model parameters. Although the potential palette of relevant dynamic models is large and could include e.g.\ morphological development models \citep{MS20} or metabolic models \citep{NBR19}, evolutionary biologists have often considered gene regulatory network models based on the framework proposed by \citealp{Wag94} as a good compromise between complexity and numerical tractability for studying the evolution of canalization and robustness \citep[for review]{LP12,Wan19}.

Such theoretical gene networks have been shown to display enough non-linearities (epistasis and pleiotropy) to evolve enhanced or reduced sensitivity to environmental \citep{Mas04,EMW11,EMW11b} and genetic \citep{Wag96,BS03,ALS+06,LR16} perturbations.Interesting observations suggest that environmental or genetic canalization could be correlated to other robustness properties in such models. In particular, it has been shown that network stability, the propensity of the network to maintain stable (non-cyclic) gene expressions, was positively correlated to robustness, as selection on stability alone could drive an indirect response of genetic \citep{SB02} and environmental \citep{Mas04} canalization. In contrast, \citealp{ORL18} showed that networks selected to maintain (but not converge to) an equilibrium could become both environmentally sensitive and genetically canalized, suggesting that environmental and genetic robustness could be theoretically decoupled. However, a systematic quantitative description of the pleiotropic pattern underlying different robustness components has never been considered. 

Here, I aim at extending the study of canalization in theoretical gene networks to address the multidimensional nature of robustness, by estimating the evolutionary independence of various robustness components. Five arbitrary robustness-related measurements were considered, two of them corresponding to environmental robustness (early vs.\ late disturbances), two corresponding to genetic robustness (early --- inherited --- or late --- acquired --- mutations), the last one (gene expression unstability) being related to the intrinsic stability of the expression phenotype. I will first describe the multidimensional patterns of robustness in small and random networks, and I then explore the evolutionary consequences of the pleiotropic nature of robustness through individual-based simulations. 



\citep{Wag96} \citep{Wag94} \citep{RL16} 

\citep{SB02} (Selection on stability -> evolution of genetic canalization)

\citep{Mas04} (Theoretical test of Waddington's canalization experiments. Selection to assimilate a phenotype initially disturbed ( = late env noise) work. However, selection for stability only also explains the rise of the new phenotype.)

\citep{Lec08} (relationship between density and robustness, sparse networks more robust and favord by selection if cost of complexity accounted for) 

\citep{MS09} (very general, intro, similar to deVisser + empirical examples) 

\citep{MT10} 

\citep{Kli19} (modern definitions of canalization, robustness etc). 

\citep{PP05}  (General model with no distinction between env and gen canalization; selection on canalization is weak (limited by the load). Cost of network size: mutation target) 

\citep{VCH99} (very abstract model of a neutral network. it is expected that pops evolve in more robust parts of the neutral network -> pop gen of canalization)

\citep{ALS+06} 

\citep{Wag05} (short opinion. Neutrality is not an essential features of mutations, it depends on the genetic background -> may be important for evolvability). 

\citep{KF09} 

\citep{FB15} Felix \& Barkoulas

\citep{WBB97} 

\citep{MH02} 

\citep{GW00} 

\citep{MHB+03} 

\citep{Nij02} (Robustness is a consequence of nonlinear development; robustness should be defined according to the underlying factor).

\citep{BS03}  (Evolutationary capacitance -- buffering and releasing genetic variation -- affects all genes in a network)

\citep{Fla05} (review on canalization, general concepts (epistasis, pleiotropy); empirical results) 

\citep{BPK06} 

\citep{Wad42} 

\citep{Wad59} 

\citep{GH94}

\citep{HHW03} (multilinear model + stabilizing selection; canalization for only a subset of genes -- the ones with higher mutation rates, Vm can possible increase)

\citep{BL97} (simple model of network robustness (transduction pathway), more likely than a fine-tuned system) 

\citep{Fra99} 

\citep{EM98} (Very long introduction on canalization \& assimilation. Accounts for the double effect of the environment. Canalization can evolve to be deactivated in stressful environments. Discussion about how to distinguish experimentally between different assimulation models. )

\citep{Ste02}

\citep{SK94} (Genetic canalization in Drosophla lines: stronger for traits more correlated to fitness. Discusses canalized reaction norms.)

Empirical: 

\citep{CMG+06}, 

\citep{Sch91}

\citep{SKK95} (From Drosophila inbred lines: traits correlated to fitness are more canalized. Same pattern for env and gen canalization.)

\citep{Wag00} (Testing the origin of robustness (yeast): duplicated genes vs interactions. Interactions dominate). 

\section{Model and Methods}

\subsection{Gene regulatory network}

The network model is derived from \cite{Wag94,Wag96} and belongs to the family of GRN sometimes refered to as "Wagner model" (see \citet{FP15} for a historical record). The main difference with the original Wagner model is that the network output (gene expressions) is quantitative and not qualitative, in the same way as in \cite{SB02}. 

More specifically, the structure of a $n$-gene network is encoded as a $n\times n$ matrix $\bm W$, while the state of the network is stored into a vector of size $n$, $\bm S$. In this setting, $W_{ij}$ represents the influence of gene $j$ on the expression of gene $i$, $W_{ij} < 0$ represents a negative interaction (inhibition), $W_{ij} > 0$ a positive interaction (activation), and $W_{ij} = 0$ denotes the absence of regulatory interaction. $S_i$ is the expression of gene $i$, ranging between 0 (no expression) and 1 (maximum expression). 

The properties of these gene networks are explored in a discrete dynamic system:
\begin{equation}
 \bm P_{t+1} = F(\bm W \bm P_t),
\end{equation}
\noindent where the function $F$ is a vectorized version of a sigmoid scaling function: $F(x_1, x_2, \dots, x_n) = [f(x_1), f(x_2), \dots, f(x_n)]$ with
\begin{equation} \label{eq:fx}
f(x) = \frac{1}{1+ \lambda_a e ^{- \mu_a x}}, 
\end{equation}
\noindent with $\lambda_a = (1-a)/a$ and $\mu_a = 1/a(1-a)$. The function $f$ is scaled such that $f(0) = a$ and $df/dx|_{x=0}=1$; the parameter $a$ thus stands for the constitutive gene expression (the expression of a gene in absence of regulators), and this function defines the scale of the matrix $\bm W$, as a small effect $W_{ij} = \delta$ on an otherwise non-regulated gene leads to the same change in gene expression $S_i = a+\delta$. 
%Setting $a=0.5$ leads to $f(x) = 1/(1+e^{-\mu x})$ with $\mu = 4$, which can be directly compared with e.g.\ \cite{PBF12}. 

Gene networks dynamics start from an initial expression $S_0$, and gene expression was updated for $T$ timesteps. By default, $P_0 = (a, a, ..., a)$, since this step immediately follows a virtual initial state with no expression. The expression phenotype corresponding to a gene network was determined by averaging gene expressions during the last $\tau$ timesteps for each gene $i$: $\bar P_i = \sum_{t=T-\tau}^T P_{it}$. 

\subsection{Robustness indicators}

Five robustness indicators were calculated, corresponding to five different aspects of genetic or environmental robustness in a gene network: network stability $\stability$, robustess to early ($\earlyenv$) and late ($\lateenv$) environmental disturbance, and robustness to early ($\earlymut$) and late ($\latemut$) genetic disturbance. All indicators were expressed on a scale homogeneous to log variances in gene expressions.

Dynamic systems based on the Wagner model often tend to generate limit cycles and never converge to a stable equilibrium. Network stability $\stability$ quantifies the capacity for a specific network to lead to stable gene expressions.  For consistency with other indicators, this unstability was measured as the log squared difference between the average expression during the last $\tau$ timesteps, and an extra timestep: $\stability_i = \log( \bar P_i - P_{T+1})^2)$. 

The robustness to early environmental disturbance $\earlyenv$ measures the capacity of a network to reach a consistent final state starting from different initial gene expressions. In practice, $R$ replicates of the network dynamics were run, in which the initial gene expressions ($P_0$) were drawn into Gaussian ($\mu = a, \sigma = \sigma_e$) distributions. The environmental robustness for each gene $i$ was measured as the log variance in the final gene expression across these replicates: $\earlyenv_i = \log [ 1/(R-1) \sum_{r=1}^R (\bar P_{i,r} - (1/R) \sum _r \bar P_{i,r})^2 ]$. 

The robustness to late environmental disturbance $\lateenv$ measures the capacity of a network to recover its equilibrium state after having being disturbed. Gene expressions after $T$ timesteps were disturbed by adding a random Gaussian noise of standard deviation $\sigma_E$ to each gene of the network, and $\lateenv_i$ was computed for each gene $i$ as the log variance in gene expression at time step $T+1$ over $R$ replicates: $\lateenv_i = \log [ 1/(R-1) \sum_{r=1}^R (P_{i,T+1,r} - (1/R) \sum_r P_{i,T+1,r})^2)]$. 

The robustness to early mutations $\earlymut$ measures the system robustness to inherited genetic mutations (modifications of the $\bm W$ matrix). A random non-zero element of the $\bm W$ matrix was shifted by a random Gaussian number of standard deviation $\sigma_m$, and its consequences on the mean expression of all network genes was recorded. The procedure was replicated $R$ times, and the robustness score $\earlymut_i$ for each gene $i$ was calculated as the log variance of gene expression across $R$ replicates: $\earlymut_i = \log[ 1/(R-1) \sum_{r=1}^R (\bar P_{i,r} - (1/R)\sum_r \bar P_{i,r})^2]$. 

Finally, the robustess to late mutations $\latemut$ measured the effect of mutations in the gene network $\bm W$ after having reached the final state. In practice, the $\bm W$ matrix was mutated in the same way as for $\earlymut$, but its consequences on gene expression were calculated for only one timestep, starting from the final state of the network. The procedure was simulated $R$ times, $\latemut_i = \log[ 1/(R-1) \sum_{r=1}^R (P_{i,T+1,r} - (1/R) \sum_r P_{i,T+1,r})^2 ]$.

All these scores were calculated for every gene $i$ of a given network, and then averaged over all genes in order to get a series of summary network descriptors 

\subsection{Random networks}

Random networks were generated as $n\times n$ matrices filled with independent identically-distributed random numbers drawn into a Gaussian ($\mu_0, \sigma_0$) distribution. The parameters of the distribution were determined so that gene expressions $P_i$ was as uniformy distributed as possible between minimum ($P=0$) and maximum ($P=1$) expressions; this corresponded to $\mu_0=-0.2$ and $\sigma_0=1.2$ in a full-density 6-gene networks, A density parameter $1/n \leq d \leq 1$ could be specified, corresponding to the frequency of non-zero slots in the $\bm W$ matrix. Zeros were placed randomly, with the constraint that all genes should be regulated by at least another one. 

\subsection{Exhaustive exploration of 2-gene networks}

The main interest of gene-network models is the complexity and the richness of the underlying genotype-phenotype. As a side effect, such models are in general difficult to handle mathematically (\cite{CTH11,LP12}). Excluding the one-gene self-regulating case (which already has non-trivial mathematical properties), the simplest network (2-by-2 matrix) has four genetic parameters, which makes the exploration of the parameter set tedious. Here, the number of dimensions was restricted by considering the set of networks that lead to a predefined arbitrary equilibrium, $\bm P^\ast_\infty = (P^\ast_1, P^\ast_2)$. As $F(\bm W \bm P^\ast_\infty) = \bm P^\ast_\infty$, the $\bm W$ matrix can be reduced to two independent parameters, $W_{11}$ and $W_{21}$:
\begin{equation}
    \bm W = F \left [\begin{pmatrix} W_{11} & A \\ W_{21} & B \end{pmatrix}  \begin{pmatrix} P^\ast_1 \\ P^\ast_2 \end{pmatrix} \right] = \begin{pmatrix}P^\ast_1 \\ P^\ast_2 \end{pmatrix},
\end{equation}
\noindent with
\begin{equation}
	\label{eq:predef}
    \begin{split}
        A = \frac{1}{P^\ast_2} [f^{-1}(P^\ast_1)-W_{11}P^\ast_1], \\
        B = \frac{1}{P^\ast_2} [f^{-1}(P^\ast_2) - W_{21} P^\ast_1],
    \end{split}
\end{equation}
\noindent $f^{-1}(x) = -\frac{1}{\mu_a} \log \left( \frac{1-x}{\lambda_a x} \right)$ being the inverse of $f(x)$ (equation~\ref{eq:fx}). The $\bm W$ matrix achieving the desired $\bm P^\ast_\infty$ equilibrium from a specific pair $W_{11}, W_{21}$ always exists (and is unique), but the stability of the equilibrium is not guaranteed. Networks which final gene expression $\bm P = (P_1,P_2)$ differed substantially from the target (in practice, when $|P_1 - P^\ast_1| + |P_2 - P^\ast_2| > 0.15$) were excluded from the analysis. Such discrepancies correspond to either unstable equilibria (in which case gene expressions were driven away from the equilibrium until reaching an extreme (0 or 1) expression) or extreme oscillatory behaviors (large oscillations may hit extreme expression limits (0 or 1) which drives the average expression away from the target equilibrium). 

\subsection{Evolutionary simulations}

The evolution of gene networks under various evolutionary constraints was studied by individual-based simulations. Each individual was featured by its genotype (a $n \times n$ $\bm W$ matrix), its expression phenotype $\bar P$, and the five robustness scores $\stability$, $\earlyenv$, $\lateenv$, $\earlymut$, and $\latemut$. For simplicity, individuals were haploid and reproduced clonally. Mutations consisted in adding a random Gaussian deviate of variance $\sigma_m^2$ to an element of the $\bm W$ matrix, with a rate $\mu$. Generations were non-overlapping, and population size $N$ was constant. A generation consisted in sampling $N$ new individuals among the $N$ parents, with a probability propotional to the individual fitness. Fitness was computed assuming stabilizing selection around a target (optimal) expression level for $n^\prime \leq n$ genes of the network, as $w = \exp(- \sum_{i=1}^{n^\prime} s_i (\bar P_i - \theta_i)^2 )$, where $s_i$ was the strength of stabiizing selection on gene $i$ ($s_i = 0$ standing for no selection), and $\theta_i$ was the optimal expression phenotype. In most cases, the $\theta_i$ were drawn in uniform (0,1) distributions at the beginning of each replicated simulation. 

In addition, direct, directional selection on robustness indicators was performed in some simulations, consisting in multiplying individual fitness by $\exp ( \sum_{x \in (S, E, e, M, m)} \beta_x \rho _x )$, where $\beta_x$ is the strength of directional (positive or negative) selection on robustness index $x$. 

Simulations and data analysis were coded in R \citep{R20}, except for the gene network dynamics that was coded in C++ and embedded in the R code with the Rcpp package \citep{EF11}. 

\subsection{Evolvability}

Evolvabilities and conditional evolvabilities of robustness indicators were computed from the mutational matrices. In practice, for a given genotype $\bm W$, a mutant genotype $\bm W^\prime$ was generated by applying two random mutations at random positions in the network interaction matrices, drawing mutational effects in a Gaussian distribution of variance $\sigma_m^2$ = 0.1. The procedure was applied 10000 times, and for each mutant genotype, the five robustness parameters as well as equilibrium gene expressions ($\bar P$) were computed. The resulting variance-covariance matrix $\bm M$ was then used to quantify the evolvability of robustness parameters; the diagonal stands for unconstrainted evolvabilities, while conditional evolvabilities were calculated as proposed in \citet{HH08} as $c_{\rho_x} = (\bm \beta_x^{-1} \bm M_Y^{-1} \bm \beta_x)^{-1}$, where $\bm M_Y$ is a subset of the matrix $\bm M$ containing the focal index $x$ and all conditioning traits $Y$, and $\bm \beta_x$ indicates the direction of selection (which is, in our case, in the direction of the focal index $x$, e.g. $\beta_x = (1,0,0,0...)$ if $x$ is the first trait. Conditionning was performed either on the expression of $n^\prime$ selected genes ($Y = (\bar P_1, \dots, \bar P_{n^\prime})$), and on the four other robustness indexes (e.g., when focusing on $\stability$, $Y = (\earlyenv, \lateenv, \earlymut, \latemut)$). 

\section{Results}

\subsection{Random networks}

Random interaction matrices are regularly used in the literature to study the general properties of gene networks (e.g.\ \cite{CTH11,PBF12}). As such, random networks are not expected to reflect the properties of biologically-realistic genetic architectures, as biological networks are far from random. However, such an approach helps developing a general intuition about the properties of the underlying model, especially about the amount of selection it takes to bring a network into its non-random state. 

Correlations were calculated between all five robustness components over 10,000 random networks (\SupMat~\ref{supp:fullcorr}). All robustness components were positively correlated. However, correlations range from above 0.9 (early vs.\ late mutations) to about 0.2 (stability vs early mutations). A Principal Component Analysis (Figure~\ref{pca}) confirms that robustness components are not tighly associated. The first PC (57\% of the total variance) corresponds to the general robustness of the network, and involves all robustness indexes. The remaining variance is explained by orthogonal vectors separating all other robustness components. Although weak (about 2\% of the total variance), the last PC still represents an independent direction of variation and is not an artifact of the sampling effect (\SupMat~\ref{supp:samplingPC}). 

\begin{figure}[t]
\begin{center}
\includegraphics[width=10cm]{../results/figB-random}
\caption{\color{Gray} \label{fig:pca} Summary of the principal component analysis on the five robustness components over 10,000 random 6-gene networks ($\mu_0=0, \sigma_0=1$). Left: position of the five robustness components on all five (normalized) Principal Components (PC); $\stability$: Stability, $\earlyenv$: Early environmental, $\lateenv$: Late environmental, $\earlymut$: Early genetic, $\latemut$: Late genetic. Right: relative contribution of the five PCs to the total variance.}
\end{center}
\end{figure}

\subsection{2-gene networks}

In the following, I considered the particular case of $\bm P_\infty$ = (0.3, 0.6) with two intermediate, distinct gene expressions. Equivalent results could be achieved with a different target. Figure~\ref{fig:imgpanels} illustrates how the robustness components vary in this constrained 2-gene network model (red stands for maximum robustness, i.e.\ minimum scores for $\stability$, $\earlyenv$, $\lateenv$, $\earlymut$, and $\latemut$). In the white regions, the equilibrium was not achieved in numerical simulations for at least three different reasons: (i) flutuations around the equilibrium are large enough to hit the edges of the (0,1) interval, shifting the mean expression; (ii) the expression dynamics was slow and the network was unable to get close to the equilibrium after 16 time steps; (iii) the equilibrium was not reachable from the default starting point.

\begin{figure}[t]
\begin{adjustwidth}{-1in}{0in}
\begin{flushright}
\includegraphics[width=15cm]{../results/figC}
\caption{\color{Gray} \label{fig:imgpanels} Robustness indicators ($\stability$, $\earlyenv$, $\lateenv$, $\earlymut$, $\latemut$) estimated for an exhaustive continuum of two-gene networks with an arbitrary expression equilibrium at $\bm P_\infty = (0.3, 0.6)$. Although two-gene networks have four independent genetic parameters, only two are represented here, the two others being computed to ensure the desired equilibrium. Red stands for the maximum robustness (lowest scores); yellow for minimum robustness (highest scores). For readability, color scales are different across panels. Letters A to E stand for five example networks illustrated in \SupMat~\ref{supp:simpanels}.}
\end{flushright}\end{adjustwidth}
\end{figure}

It immediately appears that the different robustness components are correlated, but do not overlap. The simulated behavior of networks A to E are illustrated in \SupMat~\ref{supp:simpanels}, and the corresponding $\bm W$ matrices are provided in \SupMat~\ref{supp:W}. The network denoted as "B" is robust to all five components, while network E is sensitive to all components except stability. Network C is unstable, but remains relatively canalized. Networks A and D illustrate intermediate decanalization behaviors, through different mechanisms (unstability for network D, and weak buffering for network A). 

This 2-gene network analysis thus confirms the results obtained for larger gene networks. Robustness is thus not a feature of large and intricate genetic architectures, as it is already present (and multidimensional) in the simplest gene networks. The different robustness components are partially independent, even in small and highly constrained networks. All the networks considered here converge to the same gene expression at time step 20, and can thus be considered as phenotypically equivalent ; the colored space in Figure~\ref{fig:imgpanels} thus represents a connected neutral network in which populations could evolve, and thus change the topology and the robustness of the gene network, while keeping the expression phenotype constant. 

\subsection{Evolution and evolvability of robustness}

The evolution of robustness was studied through individual-based simulations, in which all individuals were characterized by their genotype (a 6-gene network) and a set of phenotypes (gene expressions and network robustness). Gene expressions for 3 out of 6 genes were under stabilizing selection. In addition to stabilizing selection on gene expression, robustness indicators were directly selected towards more or less sensitivity. In this setting, even in the control simulations, gene networks tend to evolve towards more robustness to mutations (lower scores for $\earlymut$, Figure~\ref{fig:evol}, black circles), as a result to the evolution of genetic canalization (mutations that have a large effect on equilibrium gene expressions and stability are penalized), which prevents genetic drift to drive gene expressions to 0 or 1 for all genes. 

Direct selection on almost all robustess components lead to a response, showing that robustness is evolvable. The only exception was the robustness to late mutations $\latemut$, which could hardly evolve towards more sensitivity (i.e., the starting genotype was already very sensitive to late mutations). Yet, the evolutionary potential of the robustness indicators differed substantially, as indicated by the differences in the Y-scales. Robustness indicators being all homogeneous to a variance of gene expression, they could be compared directly. The most evolvable robustness components are Stability ($\stability$) and early environmental disturbances ($\earlyenv$), which can differ by up to 25 log units (11 orders of magnitude). In contrast, robustness to late environmental noise $\lateenv$ and genetic changes ($\earlymut$ and $\latemut$) only differed by 3 to 4 log units (i.e.\ a factor 10 to 100) after 10,000 generations of selection, although the selection response was still ongoing. 

Selection on all robustness components also lead to an indirect response of all other components, which confirms a general genetic correlation. Direct selection on environmental canalization or gene expression stability can thus induce the evolution of genetic canalization, and \emph{vice versa}. Indirect responses were generally smaller or similat to direct repsonses, except when selecting for more sensitivity for early mutations $\earlymut$. This could a be consequence from the canalizing selection on gene expression, which opposes the direct selection towards decanalized networks. 


\begin{figure}[t]
\begin{adjustwidth}{-1in}{0in}
\begin{flushright}
\includegraphics[width=15cm]{../results/figG}
\caption{\color{Gray} \label{fig:evol} The evolution of all five robustness indicators (five figure panels) was recorded for 10,000 generations in individual-based simulations. The figures show the average robustness over 20 replicated simulations. The control simulations (black circles) correspond to stabilizing selection only. Colored symbols correspond to simulations in which various robusntess indicators were selected up or down (upward or downward triangles). Bold-faced symbols indicate the result of direct selection (the robustness indicator displayed in the panel was directly selected), while thin symbols stand for indirect selection (the response is the result of selection on another indicator, which can be identified based on its color). }
\end{flushright}\end{adjustwidth}
\end{figure}

These differences in selection responses can be attributed to differences in evolvability (Figure~\ref{fig:evolvability}). Stability ($\stability$) and early environmental robustness ($\earlyenv$) display large levels of mutational evolvability, even when conditioned to gene expression (i.e.\, $\stability$ and $\earlyenv$ can evolve without affecting equilibrium gene expression, and thus the individual fitness). On the opposite, genetic robustness ($\earlymut$ and $\latemut$) display very low levels of evolvability, even when not conditioned to gene expression. The evolvability of robutness is always low when conditioned over the other indicators. Differences in evolvabilities among indicators is much reduced when comparing random networks instead of evolved networks, suggesting that the lack of evolvability of genetic robustness is largely due to the evolution of canalization, which already indirectly selects for genetic robustness under phenotypic stabilizing selection. 

\begin{figure}[t]
\begin{center}
\includegraphics[width=15cm]{../results/figJ-evolved}
\caption{\color{Gray} \label{fig:evolvability} Estimation of the evolvability (mutational variance) of all five robustness indicators in gene networks at mutation-selection-drift equilibrium (average genotype at genetation 10,000 in populations under stabilizing selection, corresponding to black circles in Figure~\ref{fig:evol}). The figure also displays conditional evolvability over gene expression (i.e.\ the possibility to change robustness without affecting fitness under stabilizing selection on gene expression), and conditional evolvability relative to the other four robustness indicators (possibility to evolve one robustness indicator without affecting the others). }
\end{center}
\end{figure}


\section{Discussion}

Whether or not various robustness components of genetic architectures are independent is a key issue to understand why organisms are robust or sensitive to genetic or environmental disturbances. In particular, independent genetic bases of robustness components would call for independent evolutionary histories, while a pleiotropic genetic architecture could explain the evolution of nonadaptive robustness components as a result of indirect selection. The analysis of the genetic correlations between five robustness components, based on a simple gene network model, results in a balanced answer: robustness components appear to be generally correlated, but the pleiotropy is not an absolute constraint, and at least some pairs of robustness components (such as the robustness to mutations and the robustness to environmental disturbances) display some genetic variation in the direction that would dissociate both variables. Such a quantitative answer to the so-called 'congruence' hypothesis \citep{dHW+03} would explain both how unselected robustness components could be partly driven by indirect selection and why various robustness-related features seem to have their own evolutionary history. 

\subsection{Model}

Robustness (and the possibility for robustness to evolve) emerge from non-linearities in the physiological or developmental processes \citep{Nij02}, which generate potentially complex epistatic and pleiotropic effects. Gene regulation networks are popular candidates when attempting to model complex biological processes: they are at least partly built on solid and realistic principles (transcription factors can enhance or repress the expression of other genes), gene regulation plays a crucial role in most biological, physiological, and developmental mchanisms, and even modest-sized regulation networks display a wide diversity of possible outcomes, including homeostasis (stable equilibrium of gene expressions), cyclic dynamics, or amplification of a weak signal. Finally, the phenotypic level considered as the output of a gene network (the expression level of all network genes) can be assimilated to a partial transcriptome, which opens the possibility for confrontation with empirical data. 

The framework proposed by \citep{Wag94,Wag96} is particularly popular in evolutionary biology to model gene network evolution (cite review Yang?), due to its computational simplicity and efficiency, combined with an obvious biological interpretation (each line of the regulation matrix is the set of transcription factor fixation sites in the promoter of a gene, cite Masel). In practice, multiple variants based on this original model have been derived, either to address specific questions, or to correct for uncomfortablty unrealistic features. Here, I used a quantitative version of the model, in which gene expressions are scaled between 0 (no expression) and 1 (maximum expression), which was first proposed in \citep{Wag94}, although later work have often preferred  binary networks (in which genes can be on/off, e.g. \citep{Wag96}, Ciliberti et al), and a gene expression scaling between -1 and 1. Unlike in Wag96, SB03, etc., mutational effects were correlated (the value of the mutant allele is drawn in a Gaussian centered around the value of the parental allele), which allows for cumulative evolution. Finally, the sigmoid response function was made asymetrical by introducing a constitutive expression parameter (as in e.g. RL16) in order to avoid the unrealistically high expression of unregulated genes (half the maximum expression) from the default Wagner model. 

Discrete time and simple matrix algebra is necessary to run evolutionary individual-based simulations, in which the network output needs to be calculated for thousands of individuals and thousands of generations. Using more realistic models based on continuous time and differential equations, non-linear regulation effects, and independent degradation and transcription rates would make the simulations impractical, with little benefit in terms of explanatory power. Computational constraints also limit the network size to a few dozen genes max, which is not enough to generate realistic levels of sparcity -- gene networks are too dense to be realistic. It is also necessary to note that the simulated phenotypic target (maintaining a constant set of gene expressions) were extremely simple compared to what such gene networks are able to do (e.g. converging to different equilibria in different cell types, or controling a complex dynamic of gene expression during the development). However, the general conclusions of this study are (at least qualitatively) robust to most simulation parameters (sup mat), suggesting that they reflect general properties of the underlying genetic architecture. 

\subsection{Robustness indicators}

There are potentially many ways to measure the robustness of a phenotypic trait. Here, five indicators were measured, in order to catch various (and potentially independent) aspects of what is generally defined as robustness. The sensitivity to inherited mutations ($\earlymut$) is probably the most popular one, as it is central to the discussion around the evolution of canalization (Wadd, Wagner). The sensitivity to environmental perturbations is also unavoidable, although its implementation in a gene network model is less straightforward. Here, it is calculated as both the sensitivity of the network to large disturbances of the initial expression state ($\earlyenv$), which measures the width of the basin of attraction of the optimal expression pattern, and as the strength of the stability of the equilibrium when slightly disturbed ($\lateenv$). These two measurements can easily be interpreted as developmental robustness and physiological homeostasis, respectively, as they quantify the response of the network to disturbances in the expression levels of different magnitudes and at different time scales. The robustness to mutations occuring after the network convergence ($\latemut$) was considered because it sets up an alternative to the genetic vs.\ environmental congruence hypothesis: in long-lived organisms, is it not unrealistic to imagine that non-heritable (somatic) mutations participate to the agieng process, and thay delayed ageing could be under direct selection. Thus, the robustness to somatic mutations could also drive indirectly the evolution of genetic canalization. Finally, the gene network stability ($\stability$, amplitude of the fluctuations of gene expressions) was considered as a robustness components because it has been proven to drive an indirect response of genetic canalization, based on very similar model (Siegal and Bergman). 

These indicators were chosen based on the possibility to measure them in numerical simulations. Although the empirical assessment of the correlation between robustness components would be way more convincing than a theoretical study, defining similar measurements from experimental datasets could be challenging. For instance, $\earlymut$ and $\earlyenv$ could, at least in theory, be estimated as the variance in gene expression across genetic backgrounds or across environmental conditions, respectively. Measuring $\latemut$ environmentally could be more complicated, as it would likely be confounded with other ageing mechanisms. In contrast, the empirical distinction between e.g.\ $\lateenv$ and $\stability$ relies on discriminating internal vs.\ external sources of noise, and might be in practice impossible. In all cases, gene expression data are generally quite noisy and their analysis necessitate heavy corrections to prevent multiple testing issues. Although the detection of modest differences in variances across conditions or genotypes may require unrealistically large high-quality data sets, this issue may vanish with the development of sequencing technology. 

Some popular measurements of developmental robustness were not considered here for technical reasons. For instance, fluctuating asymetry (the variance between the same phenotypic trait measured in the right and the left body parts of symmetric organisms) is a convenient measurement of microenvironmental effects on the development (Debat and David?), but it has no equivalent at the level of gene expression in a regulation network. The deterministic sensitivity to a directional environmental gradient could also be used to measure phenotypic plasticity, which is central to the question of phenotypic robustness. Yet, there are several ways to model phenotypic plasticity in a gene network (the environmental signal may come through the initial state of the network (Masel?), as a constant bias in the expression of some siganling genes (Odorico?)), and it requires a specific selection setup (different expression optimums as a function of the environment). Because of this additional complexity, phenotypic plasticity was excluded from the focus of this work, although the evolution of plasticity of gene expression remains an intriguing and fundamental question. In particular, phenotypic plasticity (i.e. an adaptive lack of robustness to some environmental signal) may itself be canalized (to genetic or other environmental disturbances). This paradox could have played a major role in Waddington's refusal to split canalization into genetic and environmental components (cite?), and is far from being elucidated now, as it would suggest that robustness of robustness may be both variable and evolvable (which is, arguably, unpleasant).  



\printbibliography




\clearpage



\begin{appendices}
  \setcounter{section}{0}
  \renewcommand{\thesection}{\arabic{section}}

  \clearpage
  \section{}
    \label{supp:fullcorr}
    \subsection*{Correlations among robustness indexes}
	\begin{adjustwidth}{-1in}{0in}
	\begin{flushright}
	\includegraphics[width=15cm]{../results/figA}
	\end{flushright}

	\color{Gray} Correlations between all five robustness components among 10,000 random 10-gene networks ($\mu_0=-0.2, \sigma_0=1.2$). 
	\end{adjustwidth}


  \clearpage
  \section{}
    \label{supp:W}
    \subsection*{Two-gene example networks}

	\begin{center}
	\begin{tabular}{rrrrr}
	  \hline
	 & $W_{11}$ & $W_{21}$ & $W_{12}$ & $W_{22}$ \\ 
	  \hline
	  A & 0.70 & 0.20 & -0.21 & 0.38 \\ 
	  B & -0.30 & 0.30 & 0.29 & 0.33 \\ 
	  C & -0.40 & 0.80 & 0.34 & 0.08 \\ 
	  D & -1.00 & -0.80 & 0.64 & 0.88 \\ 
	  E & 1.50 & 3.50 & -0.61 & -1.27 \\ 
	   \hline
	\end{tabular}
	\end{center}
	
	{\color{Gray} The five two-gene networks detailed in figures~\ref{fig:imgpanels} and~\ref{fig:simpanels}.}

  \clearpage
  \section{}
    \label{supp:whyitfails}
    \subsection*{Reasons for not reaching the desired equilibrium}
	\begin{center}
	\includegraphics[width=6cm]{../results/figCs}
	\end{center}
	
	{\color{Gray} Although equation~\ref{eq:predef} guarantees that an equilibrium exists at the target phenotypic expression, the equilibrium might not reachable in practice when simulating the gene network dynamics. The colored area in the figure corresponds to networks that failed to produced the target phenotype, each color representing a distinct reason; Yellow: network dynamics was slow and the final gene expression has not been reached yet after 16 time steps; Gray: an alternative equilibrium was reached (most of the time implying that one or both genes are either completely silenced to fully expressed). Red: The network steady state featured oscillations that were so large that they hit the maximum or minimum expression, shifting the average expression away from the target expression. }

  \clearpage


  \clearpage
  \section{}
    \label{supp:simpanels}
    \subsection*{Illustration of the robustness scores}

	\begin{adjustwidth}{-1in}{0in}
	\begin{flushright}
	\includegraphics[width=15cm]{../results/figD}
	\end{flushright}
	\color{Gray} The figure displays a subset of the replicated tests for four robustness indexes. Columns A to E correspond to the five networks described in \SupMat~\ref{supp:W}. In each panel, the default (undisturbed) network kinetics is displayed as plain lines (black for gene~1, red for gene~2). By construction, all networks have an equilibrium at (0.3, 0.6). The network stability can be assessed from the amplitude of the cycles in the undisturbed kinetics. Network homeostasis (first line) was estimated by disturbing the expression state of the genes (for readability, the network response was expanded over several time steps). Environmental canalization (second line) was estimated by disturbing the initial state (semi-transparent lines). Genetic canalization (third line) was measured as the variance in the final state when mutating the network at the beginning of the development. Finally, the robustness to late mutations (last line) consisted in mutating the network after having reached the equilibrium.
	\end{adjustwidth}


  \clearpage
  \section{}
    \label{supp:sensitmag}
    \subsection*{Sensitivity of the robustness measurements to the magnitude of the disturbance}
	\begin{center}
	\includegraphics[width=7.5cm]{../results/figM-random} \includegraphics[width=7.5cm]{../results/figM-evolved}
	\end{center}
	{\color{Gray} Four out of five robustness indicators ($\earlyenv$, $\lateenv$, $\earlymut$, $\latemut$) depend on the magnitude of the disturbance. The figure displays the influence of the size of the disturbance on the robustness measurement (left: on random networks, right: on evolved networks). The vertical lines stand for the disturbance size used in this paper. Overall, all robutness indexes tend to increase along with the disturbance. There is a substantial difference in the relationship between the disturbance and the robustness score between random and evolved networks; robustness scores appear very reliable for evolved networks (the rank of different genotypes in terms of robustness does not depend on the size of the disturbance), while the picture is more messy for random networks, which can be differentially robust to large or small disturbances. }


  \clearpage
  \section{}
    \label{supp:explo}
    \subsection*{Exploration of the parameter set}
	\begin{center}
	\includegraphics[width=12cm]{../results/figL} 
	\end{center}
	
	{\color{Gray}Influence of various simulation parameters (mutation rate $\mu$, population size $N$, total number of genes $n$, number of selected genes $n^\prime$, and strenght of selection $s$) on fitness and robustness indexes at equilibrium. The figure reports the mean $\pm$ standard deviation across 20 replicated simulations. Vertical dotted lines stand for the default parameter values in the simulations. }




\end{appendices}

\end{document}
